{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_id = \"\"\"query Search($dataset: DatasetId!, $query: String!) {\n",
    "    searchResults(dataset: $dataset, query: $query) {\n",
    "      label\n",
    "      value: url\n",
    "    }\n",
    "  }\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "query_variables = \"\"\"{\n",
    "  \"dataset\": \"gnomad_r3\",\n",
    "  \"query\": \"%s\"\n",
    "}\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pynomad.Search import Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "sch = Search('gnomad_r3', gene_id, query_variables)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "ans = sch.request_gnomad(\"idua\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/gene/ENSG00000127415?dataset=gnomad_r3'"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "val = ans['data']['searchResults'][0]['value']\n",
    "val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "ensid = val.split('/')[-1].split('?')[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "class Search:\n",
    "\n",
    "    def __init__(self, dataset_id, query, query_variables):\n",
    "        self.ds = dataset_id\n",
    "        self.end_point = \"https://gnomad.broadinstitute.org/api/\"\n",
    "        self.query = query\n",
    "        self.query_vars = query_variables\n",
    "\n",
    "        self.dm = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "class GeneSearch(Search):\n",
    "\n",
    "    def __init__(self, gene: str):\n",
    "        super().__init__(\"aro\", \"\", \"\")\n",
    "        self.gene = gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "fs = GeneSearch(\"aa\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'aro'"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fs.ds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pynomad import GeneSearch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gs = GeneSearch(3, \"idua\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ENSG00000127415'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gs.gene_ens_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Trp402Ter</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>[]</td>\n",
       "      <td>120</td>\n",
       "      <td>142464</td>\n",
       "      <td>0.000842</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-1002750-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>141870</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002750</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-1002751-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143004</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002751</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-1002756-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val405Gly</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>141746</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002756</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-1002759-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser406Trp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>142602</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002759</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4-1002766-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala408Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>143012</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002766</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4-1002767-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Arg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2750</td>\n",
       "      <td>142238</td>\n",
       "      <td>0.019334</td>\n",
       "      <td>67</td>\n",
       "      <td>4</td>\n",
       "      <td>1002767</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-1002768-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Glu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143002</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4-1002770-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Thr410Pro</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>140858</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4-1002770-AC-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val411SerfsTer29</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>140860</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant ID  Gene         Consequence          Annotation Flags  \\\n",
       "0   4-1002747-G-A  IDUA         p.Trp402Ter         stop_gained    []   \n",
       "1   4-1002750-C-T  IDUA         p.Ala403Val    missense_variant    []   \n",
       "2   4-1002751-C-G  IDUA         p.Ala403Ala  synonymous_variant    []   \n",
       "3   4-1002756-T-G  IDUA         p.Val405Gly    missense_variant    []   \n",
       "4   4-1002759-C-G  IDUA         p.Ser406Trp    missense_variant    []   \n",
       "5   4-1002766-C-T  IDUA         p.Ala408Ala  synonymous_variant    []   \n",
       "6   4-1002767-G-C  IDUA         p.Gly409Arg    missense_variant    []   \n",
       "7   4-1002768-G-A  IDUA         p.Gly409Glu    missense_variant    []   \n",
       "8   4-1002770-A-C  IDUA         p.Thr410Pro    missense_variant    []   \n",
       "9  4-1002770-AC-A  IDUA  p.Val411SerfsTer29  frameshift_variant    []   \n",
       "\n",
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0           120         142464          0.000842                      0   \n",
       "1             2         141870          0.000014                      0   \n",
       "2             2         143004          0.000014                      0   \n",
       "3             1         141746          0.000007                      0   \n",
       "4             2         142602          0.000014                      0   \n",
       "5             3         143012          0.000021                      0   \n",
       "6          2750         142238          0.019334                     67   \n",
       "7             2         143002          0.000014                      0   \n",
       "8             5         140858          0.000035                      0   \n",
       "9             1         140860          0.000007                      0   \n",
       "\n",
       "  Chromosome Location Reference Alternative  \n",
       "0          4  1002747         G           A  \n",
       "1          4  1002750         C           T  \n",
       "2          4  1002751         C           G  \n",
       "3          4  1002756         T           G  \n",
       "4          4  1002759         C           G  \n",
       "5          4  1002766         C           T  \n",
       "6          4  1002767         G           C  \n",
       "7          4  1002768         G           A  \n",
       "8          4  1002770         A           C  \n",
       "9          4  1002770        AC           A  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynomad import RegionSearch\n",
    "rs = RegionSearch(3, 4, 1002741, 1002771)\n",
    "normal, _ = rs.get_data()\n",
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "ans = gs.get_json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'4'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ans['data']['gene']['chrom']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-987000-A-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-85A&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>140350</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>987000</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-987002-A-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-83A&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>140342</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>987002</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-987003-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-82T&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>139354</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>987003</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-987004-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-81G&gt;A</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>142964</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>987004</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-987005-G-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-80G&gt;T</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>143228</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>987005</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5525</th>\n",
       "      <td>4-1004487-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*94C&gt;T</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>4</td>\n",
       "      <td>143296</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1004487</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5526</th>\n",
       "      <td>4-1004491-A-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*98A&gt;T</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>143074</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1004491</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5527</th>\n",
       "      <td>4-1004492-T-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*99T&gt;C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143280</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1004492</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5528</th>\n",
       "      <td>4-1004503-TTTA-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>None</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>25</td>\n",
       "      <td>143260</td>\n",
       "      <td>0.000175</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1004503</td>\n",
       "      <td>TTTA</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5529</th>\n",
       "      <td>4-1004505-T-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*112T&gt;C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>143248</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1004505</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5530 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Variant ID  Gene Consequence           Annotation Flags  \\\n",
       "0         4-987000-A-G  IDUA    c.-85A>G  5_prime_UTR_variant    []   \n",
       "1         4-987002-A-G  IDUA    c.-83A>G  5_prime_UTR_variant    []   \n",
       "2         4-987003-T-G  IDUA    c.-82T>G  5_prime_UTR_variant    []   \n",
       "3         4-987004-G-A  IDUA    c.-81G>A  5_prime_UTR_variant    []   \n",
       "4         4-987005-G-T  IDUA    c.-80G>T  5_prime_UTR_variant    []   \n",
       "...                ...   ...         ...                  ...   ...   \n",
       "5525     4-1004487-C-T  IDUA    c.*94C>T  3_prime_UTR_variant    []   \n",
       "5526     4-1004491-A-T  IDUA    c.*98A>T  3_prime_UTR_variant    []   \n",
       "5527     4-1004492-T-C  IDUA    c.*99T>C  3_prime_UTR_variant    []   \n",
       "5528  4-1004503-TTTA-T  IDUA        None  3_prime_UTR_variant    []   \n",
       "5529     4-1004505-T-C  IDUA   c.*112T>C  3_prime_UTR_variant    []   \n",
       "\n",
       "      Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0                1         140350          0.000007                      0   \n",
       "1                1         140342          0.000007                      0   \n",
       "2                1         139354          0.000007                      0   \n",
       "3                1         142964          0.000007                      0   \n",
       "4                1         143228          0.000007                      0   \n",
       "...            ...            ...               ...                    ...   \n",
       "5525             4         143296          0.000028                      0   \n",
       "5526             1         143074          0.000007                      0   \n",
       "5527             2         143280          0.000014                      0   \n",
       "5528            25         143260          0.000175                      0   \n",
       "5529             1         143248          0.000007                      0   \n",
       "\n",
       "     Chromosome Location Reference Alternative  \n",
       "0             4   987000         A           G  \n",
       "1             4   987002         A           G  \n",
       "2             4   987003         T           G  \n",
       "3             4   987004         G           A  \n",
       "4             4   987005         G           T  \n",
       "...         ...      ...       ...         ...  \n",
       "5525          4  1004487         C           T  \n",
       "5526          4  1004491         A           T  \n",
       "5527          4  1004492         T           C  \n",
       "5528          4  1004503      TTTA           T  \n",
       "5529          4  1004505         T           C  \n",
       "\n",
       "[5530 rows x 13 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynomad import GeneSearch\n",
    "gs = GeneSearch(3, \"idua\")\n",
    "normal, clinical = gs.get_data()\n",
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>clinvar_variation_id</th>\n",
       "      <th>gold_stars</th>\n",
       "      <th>major_consequence</th>\n",
       "      <th>pos</th>\n",
       "      <th>variant_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[Uncertain_significance]</td>\n",
       "      <td>350186</td>\n",
       "      <td>1</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>987014</td>\n",
       "      <td>4-987014-C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[Pathogenic]</td>\n",
       "      <td>550458</td>\n",
       "      <td>1</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>987085</td>\n",
       "      <td>4-987085-A-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[Pathogenic]</td>\n",
       "      <td>639529</td>\n",
       "      <td>1</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>987086</td>\n",
       "      <td>4-987086-T-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[Likely_pathogenic]</td>\n",
       "      <td>557150</td>\n",
       "      <td>1</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>987087</td>\n",
       "      <td>4-987087-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[Uncertain_significance]</td>\n",
       "      <td>526832</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>987095</td>\n",
       "      <td>4-987095-T-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>383</th>\n",
       "      <td>[Uncertain_significance]</td>\n",
       "      <td>558359</td>\n",
       "      <td>1</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1004349</td>\n",
       "      <td>4-1004349-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>384</th>\n",
       "      <td>[Likely_benign]</td>\n",
       "      <td>743489</td>\n",
       "      <td>1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1004381</td>\n",
       "      <td>4-1004381-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>385</th>\n",
       "      <td>[Likely_benign]</td>\n",
       "      <td>754394</td>\n",
       "      <td>1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1004390</td>\n",
       "      <td>4-1004390-A-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>386</th>\n",
       "      <td>[Pathogenic]</td>\n",
       "      <td>11920</td>\n",
       "      <td>0</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>1004391</td>\n",
       "      <td>4-1004391-T-G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387</th>\n",
       "      <td>None</td>\n",
       "      <td>242721</td>\n",
       "      <td>0</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>1004393</td>\n",
       "      <td>4-1004393-A-T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>388 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        clinical_significance clinvar_variation_id  gold_stars  \\\n",
       "0    [Uncertain_significance]               350186           1   \n",
       "1                [Pathogenic]               550458           1   \n",
       "2                [Pathogenic]               639529           1   \n",
       "3         [Likely_pathogenic]               557150           1   \n",
       "4    [Uncertain_significance]               526832           1   \n",
       "..                        ...                  ...         ...   \n",
       "383  [Uncertain_significance]               558359           1   \n",
       "384           [Likely_benign]               743489           1   \n",
       "385           [Likely_benign]               754394           1   \n",
       "386              [Pathogenic]                11920           0   \n",
       "387                      None               242721           0   \n",
       "\n",
       "       major_consequence      pos     variant_id  \n",
       "0    5_prime_UTR_variant   987014   4-987014-C-T  \n",
       "1             start_lost   987085   4-987085-A-C  \n",
       "2             start_lost   987086   4-987086-T-C  \n",
       "3             start_lost   987087   4-987087-G-A  \n",
       "4       missense_variant   987095   4-987095-T-C  \n",
       "..                   ...      ...            ...  \n",
       "383          stop_gained  1004349  4-1004349-G-T  \n",
       "384   synonymous_variant  1004381  4-1004381-G-T  \n",
       "385   synonymous_variant  1004390  4-1004390-A-T  \n",
       "386            stop_lost  1004391  4-1004391-T-G  \n",
       "387            stop_lost  1004393  4-1004393-A-T  \n",
       "\n",
       "[388 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
