{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import json_normalize as json_normalize\n",
    "from tqdm import tqdm\n",
    "import pandas as pd\n",
    "import requests\n",
    "import argparse\n",
    "import json\n",
    "import os\n",
    "from copy import deepcopy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "query = \"\"\"\n",
    "    query GnomadVariant($variantId: String, $rsid: String, $datasetId: DatasetId!) {\n",
    "  variant(variantId: $variantId, rsid: $rsid, dataset: $datasetId) {\n",
    "    variantId\n",
    "    reference_genome\n",
    "    chrom\n",
    "    pos\n",
    "    ref\n",
    "    alt\n",
    "    colocatedVariants\n",
    "    multiNucleotideVariants {\n",
    "      combined_variant_id\n",
    "      changes_amino_acids\n",
    "      n_individuals\n",
    "      other_constituent_snvs\n",
    "    }\n",
    "    exome {\n",
    "      ac\n",
    "      an\n",
    "      ac_hemi\n",
    "      ac_hom\n",
    "      faf95 {\n",
    "        popmax\n",
    "        popmax_population\n",
    "      }\n",
    "      filters\n",
    "      populations {\n",
    "        id\n",
    "        ac\n",
    "        an\n",
    "        ac_hemi\n",
    "        ac_hom\n",
    "      }\n",
    "      age_distribution {\n",
    "        het {\n",
    "          bin_edges\n",
    "          bin_freq\n",
    "          n_smaller\n",
    "          n_larger\n",
    "        }\n",
    "        hom {\n",
    "          bin_edges\n",
    "          bin_freq\n",
    "          n_smaller\n",
    "          n_larger\n",
    "        }\n",
    "      }\n",
    "      qualityMetrics {\n",
    "        alleleBalance {\n",
    "          alt {\n",
    "            bin_edges\n",
    "            bin_freq\n",
    "            n_smaller\n",
    "            n_larger\n",
    "          }\n",
    "        }\n",
    "        genotypeDepth {\n",
    "          all {\n",
    "            bin_edges\n",
    "            bin_freq\n",
    "            n_smaller\n",
    "            n_larger\n",
    "          }\n",
    "          alt {\n",
    "            bin_edges\n",
    "            bin_freq\n",
    "            n_smaller\n",
    "            n_larger\n",
    "          }\n",
    "        }\n",
    "        genotypeQuality {\n",
    "          all {\n",
    "            bin_edges\n",
    "            bin_freq\n",
    "            n_smaller\n",
    "            n_larger\n",
    "          }\n",
    "          alt {\n",
    "            bin_edges\n",
    "            bin_freq\n",
    "            n_smaller\n",
    "            n_larger\n",
    "          }\n",
    "        }\n",
    "        siteQualityMetrics {\n",
    "          metric\n",
    "          value\n",
    "        }\n",
    "      }\n",
    "    }\n",
    "    genome {\n",
    "      ac\n",
    "      an\n",
    "      ac_hemi\n",
    "      ac_hom\n",
    "      faf95 {\n",
    "        popmax\n",
    "        popmax_population\n",
    "      }\n",
    "      filters\n",
    "      populations {\n",
    "        id\n",
    "        ac\n",
    "        an\n",
    "        ac_hemi\n",
    "        ac_hom\n",
    "      }\n",
    "      age_distribution {\n",
    "        het {\n",
    "          bin_edges\n",
    "          bin_freq\n",
    "          n_smaller\n",
    "          n_larger\n",
    "        }\n",
    "        hom {\n",
    "          bin_edges\n",
    "          bin_freq\n",
    "          n_smaller\n",
    "          n_larger\n",
    "        }\n",
    "      }\n",
    "      qualityMetrics {\n",
    "        alleleBalance {\n",
    "          alt {\n",
    "            bin_edges\n",
    "            bin_freq\n",
    "            n_smaller\n",
    "            n_larger\n",
    "          }\n",
    "        }\n",
    "        genotypeDepth {\n",
    "          all {\n",
    "            bin_edges\n",
    "            bin_freq\n",
    "            n_smaller\n",
    "            n_larger\n",
    "          }\n",
    "          alt {\n",
    "            bin_edges\n",
    "            bin_freq\n",
    "            n_smaller\n",
    "            n_larger\n",
    "          }\n",
    "        }\n",
    "        genotypeQuality {\n",
    "          all {\n",
    "            bin_edges\n",
    "            bin_freq\n",
    "            n_smaller\n",
    "            n_larger\n",
    "          }\n",
    "          alt {\n",
    "            bin_edges\n",
    "            bin_freq\n",
    "            n_smaller\n",
    "            n_larger\n",
    "          }\n",
    "        }\n",
    "        siteQualityMetrics {\n",
    "          metric\n",
    "          value\n",
    "        }\n",
    "      }\n",
    "    }\n",
    "    flags\n",
    "    rsid\n",
    "    sortedTranscriptConsequences {\n",
    "      canonical\n",
    "      gene_id\n",
    "      gene_version\n",
    "      gene_symbol\n",
    "      hgvs\n",
    "      hgvsc\n",
    "      hgvsp\n",
    "      lof\n",
    "      lof_flags\n",
    "      lof_filter\n",
    "      major_consequence\n",
    "      polyphen_prediction\n",
    "      sift_prediction\n",
    "      transcript_id\n",
    "      transcript_version\n",
    "    }\n",
    "  }\n",
    "}\n",
    "\"\"\"\n",
    "\n",
    "variables = \"\"\"\n",
    "{\n",
    "  \"datasetId\": \"%s\",\n",
    "  \"variantId\": \"%s\"\n",
    "}\n",
    "\"\"\"\n",
    "\n",
    "variables = variables % (\"gnomad_r2_1\", \"4-980801-G-A\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "end_point = \"https://gnomad.broadinstitute.org/api/\"\n",
    "response = requests.post(end_point, data={'query': query, 'variables': variables}, timeout=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = response.json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pynomad import DataManager"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>data.variant.variantId</th>\n",
       "      <th>data.variant.reference_genome</th>\n",
       "      <th>data.variant.chrom</th>\n",
       "      <th>data.variant.pos</th>\n",
       "      <th>data.variant.ref</th>\n",
       "      <th>data.variant.alt</th>\n",
       "      <th>data.variant.colocatedVariants</th>\n",
       "      <th>data.variant.multiNucleotideVariants</th>\n",
       "      <th>data.variant.exome</th>\n",
       "      <th>data.variant.genome.ac</th>\n",
       "      <th>...</th>\n",
       "      <th>data.variant.genome.qualityMetrics.genotypeQuality.all.n_smaller</th>\n",
       "      <th>data.variant.genome.qualityMetrics.genotypeQuality.all.n_larger</th>\n",
       "      <th>data.variant.genome.qualityMetrics.genotypeQuality.alt.bin_edges</th>\n",
       "      <th>data.variant.genome.qualityMetrics.genotypeQuality.alt.bin_freq</th>\n",
       "      <th>data.variant.genome.qualityMetrics.genotypeQuality.alt.n_smaller</th>\n",
       "      <th>data.variant.genome.qualityMetrics.genotypeQuality.alt.n_larger</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics</th>\n",
       "      <th>data.variant.flags</th>\n",
       "      <th>data.variant.rsid</th>\n",
       "      <th>data.variant.sortedTranscriptConsequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-980801-G-A</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>4</td>\n",
       "      <td>980801</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55,...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[{'metric': 'BaseQRankSum', 'value': 2.55}, {'...</td>\n",
       "      <td>[]</td>\n",
       "      <td>rs1165676079</td>\n",
       "      <td>[{'canonical': True, 'gene_id': 'ENSG000001274...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  data.variant.variantId data.variant.reference_genome data.variant.chrom  \\\n",
       "0           4-980801-G-A                        GRCh37                  4   \n",
       "\n",
       "   data.variant.pos data.variant.ref data.variant.alt  \\\n",
       "0            980801                G                A   \n",
       "\n",
       "  data.variant.colocatedVariants data.variant.multiNucleotideVariants  \\\n",
       "0                             []                                   []   \n",
       "\n",
       "  data.variant.exome  data.variant.genome.ac  ...  \\\n",
       "0               None                       2  ...   \n",
       "\n",
       "   data.variant.genome.qualityMetrics.genotypeQuality.all.n_smaller  \\\n",
       "0                                                  0                  \n",
       "\n",
       "   data.variant.genome.qualityMetrics.genotypeQuality.all.n_larger  \\\n",
       "0                                                  0                 \n",
       "\n",
       "   data.variant.genome.qualityMetrics.genotypeQuality.alt.bin_edges  \\\n",
       "0  [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55,...                  \n",
       "\n",
       "   data.variant.genome.qualityMetrics.genotypeQuality.alt.bin_freq  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...                 \n",
       "\n",
       "  data.variant.genome.qualityMetrics.genotypeQuality.alt.n_smaller  \\\n",
       "0                                                  0                 \n",
       "\n",
       "  data.variant.genome.qualityMetrics.genotypeQuality.alt.n_larger  \\\n",
       "0                                                  0                \n",
       "\n",
       "  data.variant.genome.qualityMetrics.siteQualityMetrics data.variant.flags  \\\n",
       "0  [{'metric': 'BaseQRankSum', 'value': 2.55}, {'...                    []   \n",
       "\n",
       "  data.variant.rsid          data.variant.sortedTranscriptConsequences  \n",
       "0      rs1165676079  [{'canonical': True, 'gene_id': 'ENSG000001274...  \n",
       "\n",
       "[1 rows x 49 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.json_normalize(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'South Asian'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "POPULATION_ID_MAP = {\n",
    "            'AFR': 'African',\n",
    "            'AMI': 'Amish',\n",
    "            'AMR': 'Latino',\n",
    "            'ASJ': 'Ashkenazi Jewish',\n",
    "            'EAS': 'East Asian',\n",
    "            'FIN': 'European (Finnish)',\n",
    "            'NFE': 'European (non-Finnish)',\n",
    "            'OTH': 'Other',\n",
    "            'SAS': 'South Asian'\n",
    "        }\n",
    "\n",
    "POPULATION_ID_MAP['SAS']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ac</th>\n",
       "      <th>an</th>\n",
       "      <th>ac_hemi</th>\n",
       "      <th>ac_hom</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AFR</th>\n",
       "      <td>0</td>\n",
       "      <td>8642</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFR_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>3654</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFR_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>4988</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMR</th>\n",
       "      <td>0</td>\n",
       "      <td>842</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMR_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>410</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMR_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>432</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASJ</th>\n",
       "      <td>0</td>\n",
       "      <td>286</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASJ_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASJ_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>208</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAS</th>\n",
       "      <td>0</td>\n",
       "      <td>1558</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAS_JPN</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAS_KOR</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAS_OEA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAS_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>538</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAS_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>1020</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIN</th>\n",
       "      <td>0</td>\n",
       "      <td>3446</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIN_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>1838</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIN_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>1608</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE</th>\n",
       "      <td>2</td>\n",
       "      <td>15334</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE_BGR</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE_EST</th>\n",
       "      <td>0</td>\n",
       "      <td>4524</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE_NWE</th>\n",
       "      <td>2</td>\n",
       "      <td>8572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE_ONF</th>\n",
       "      <td>0</td>\n",
       "      <td>2134</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE_SEU</th>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE_SWE</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE_FEMALE</th>\n",
       "      <td>2</td>\n",
       "      <td>6762</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>8572</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTH</th>\n",
       "      <td>0</td>\n",
       "      <td>1086</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTH_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTH_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>524</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAS</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAS_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAS_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FEMALE</th>\n",
       "      <td>2</td>\n",
       "      <td>13842</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>17352</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            ac     an  ac_hemi  ac_hom\n",
       "id                                    \n",
       "AFR          0   8642      0.0       0\n",
       "AFR_FEMALE   0   3654      0.0       0\n",
       "AFR_MALE     0   4988      0.0       0\n",
       "AMR          0    842      0.0       0\n",
       "AMR_FEMALE   0    410      0.0       0\n",
       "AMR_MALE     0    432      0.0       0\n",
       "ASJ          0    286      0.0       0\n",
       "ASJ_FEMALE   0     78      0.0       0\n",
       "ASJ_MALE     0    208      0.0       0\n",
       "EAS          0   1558      0.0       0\n",
       "EAS_JPN      0      0      NaN       0\n",
       "EAS_KOR      0      0      NaN       0\n",
       "EAS_OEA      0      0      NaN       0\n",
       "EAS_FEMALE   0    538      0.0       0\n",
       "EAS_MALE     0   1020      0.0       0\n",
       "FIN          0   3446      0.0       0\n",
       "FIN_FEMALE   0   1838      0.0       0\n",
       "FIN_MALE     0   1608      0.0       0\n",
       "NFE          2  15334      0.0       0\n",
       "NFE_BGR      0      0      NaN       0\n",
       "NFE_EST      0   4524      NaN       0\n",
       "NFE_NWE      2   8572      NaN       0\n",
       "NFE_ONF      0   2134      NaN       0\n",
       "NFE_SEU      0    104      NaN       0\n",
       "NFE_SWE      0      0      NaN       0\n",
       "NFE_FEMALE   2   6762      0.0       0\n",
       "NFE_MALE     0   8572      0.0       0\n",
       "OTH          0   1086      0.0       0\n",
       "OTH_FEMALE   0    562      0.0       0\n",
       "OTH_MALE     0    524      0.0       0\n",
       "SAS          0      0      0.0       0\n",
       "SAS_FEMALE   0      0      0.0       0\n",
       "SAS_MALE     0      0      0.0       0\n",
       "FEMALE       2  13842      0.0       0\n",
       "MALE         0  17352      0.0       0"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.json_normalize(data['data']['variant']['genome']['populations']).set_index('id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "def build_variant_search_standard_df(data):\n",
    "        \n",
    "        reqdf = pd.json_normalize(data['data']['variant']['genome']['populations']).set_index('id')\n",
    "        POPULATION_ID_MAP['FEMALE'] = 'Female'\n",
    "        POPULATION_ID_MAP['MALE'] = 'Male'\n",
    "\n",
    "        new_index = {}\n",
    "        frequencies = []\n",
    "        rows_to_delete = []\n",
    "        for row in reqdf.iterrows():\n",
    "            try:\n",
    "                row_id = row[0]\n",
    "                new_row_name = \"\"\n",
    "                for piece in row_id.split('_'):\n",
    "                    new_row_name += POPULATION_ID_MAP[piece] + \" \"\n",
    "                new_row_name = new_row_name[0:-1]\n",
    "                new_index[row_id] = new_row_name\n",
    "                frequencies.append(row[1]['ac']/row[1]['an'])\n",
    "            except:\n",
    "                rows_to_delete.append(row[0])\n",
    "        \n",
    "        reqdf = reqdf.drop(rows_to_delete)\n",
    "        print(reqdf)\n",
    "\n",
    "        new_index['FEMALE'] = 'Total Female'\n",
    "        new_index['MALE'] = 'Total Male'\n",
    "        new_columns = {'ac': 'Allele Count', 'an': 'Allele Number',\n",
    "                    'ac_hemi': 'Number of Hemizygotes', 'ac_hom': 'Number of Homozygotes'}\n",
    "\n",
    "        df = reqdf.rename(columns=new_columns, index=new_index)\n",
    "\n",
    "        total_row = df.loc['Total Female'] + df.loc['Total Male']\n",
    "        total_row.name = 'Total'\n",
    "        df = df.append([total_row])\n",
    "\n",
    "        frequencies.append(df.loc['Total']['Allele Count'] / df.loc['Total']['Allele Number'])\n",
    "        df['Allele Frequency'] = frequencies\n",
    "        \n",
    "        chromosome = data['data']['variant']['chrom']\n",
    "        if chromosome != 'X' and chromosome != 'Y':\n",
    "            del df['Number of Hemizygotes']\n",
    "\n",
    "        return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            ac     an  ac_hemi  ac_hom\n",
      "id                                    \n",
      "AFR          0   8642      0.0       0\n",
      "AFR_FEMALE   0   3654      0.0       0\n",
      "AFR_MALE     0   4988      0.0       0\n",
      "AMR          0    842      0.0       0\n",
      "AMR_FEMALE   0    410      0.0       0\n",
      "AMR_MALE     0    432      0.0       0\n",
      "ASJ          0    286      0.0       0\n",
      "ASJ_FEMALE   0     78      0.0       0\n",
      "ASJ_MALE     0    208      0.0       0\n",
      "EAS          0   1558      0.0       0\n",
      "EAS_FEMALE   0    538      0.0       0\n",
      "EAS_MALE     0   1020      0.0       0\n",
      "FIN          0   3446      0.0       0\n",
      "FIN_FEMALE   0   1838      0.0       0\n",
      "FIN_MALE     0   1608      0.0       0\n",
      "NFE          2  15334      0.0       0\n",
      "NFE_FEMALE   2   6762      0.0       0\n",
      "NFE_MALE     0   8572      0.0       0\n",
      "OTH          0   1086      0.0       0\n",
      "OTH_FEMALE   0    562      0.0       0\n",
      "OTH_MALE     0    524      0.0       0\n",
      "SAS          0      0      0.0       0\n",
      "SAS_FEMALE   0      0      0.0       0\n",
      "SAS_MALE     0      0      0.0       0\n",
      "FEMALE       2  13842      0.0       0\n",
      "MALE         0  17352      0.0       0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-39-d135fe0c2d22>:18: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  frequencies.append(row[1]['ac']/row[1]['an'])\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Allele Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>African</th>\n",
       "      <td>0.0</td>\n",
       "      <td>8642.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3654.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4988.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino</th>\n",
       "      <td>0.0</td>\n",
       "      <td>842.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>410.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>432.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <td>0.0</td>\n",
       "      <td>286.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1558.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>538.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1020.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3446.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1838.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1608.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <td>2.0</td>\n",
       "      <td>15334.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Female</th>\n",
       "      <td>2.0</td>\n",
       "      <td>6762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>8572.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1086.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>562.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>524.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total Female</th>\n",
       "      <td>2.0</td>\n",
       "      <td>13842.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>17352.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>2.0</td>\n",
       "      <td>31194.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000064</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Allele Count  Allele Number  \\\n",
       "African                                 0.0         8642.0   \n",
       "African Female                          0.0         3654.0   \n",
       "African Male                            0.0         4988.0   \n",
       "Latino                                  0.0          842.0   \n",
       "Latino Female                           0.0          410.0   \n",
       "Latino Male                             0.0          432.0   \n",
       "Ashkenazi Jewish                        0.0          286.0   \n",
       "Ashkenazi Jewish Female                 0.0           78.0   \n",
       "Ashkenazi Jewish Male                   0.0          208.0   \n",
       "East Asian                              0.0         1558.0   \n",
       "East Asian Female                       0.0          538.0   \n",
       "East Asian Male                         0.0         1020.0   \n",
       "European (Finnish)                      0.0         3446.0   \n",
       "European (Finnish) Female               0.0         1838.0   \n",
       "European (Finnish) Male                 0.0         1608.0   \n",
       "European (non-Finnish)                  2.0        15334.0   \n",
       "European (non-Finnish) Female           2.0         6762.0   \n",
       "European (non-Finnish) Male             0.0         8572.0   \n",
       "Other                                   0.0         1086.0   \n",
       "Other Female                            0.0          562.0   \n",
       "Other Male                              0.0          524.0   \n",
       "South Asian                             0.0            0.0   \n",
       "South Asian Female                      0.0            0.0   \n",
       "South Asian Male                        0.0            0.0   \n",
       "Total Female                            2.0        13842.0   \n",
       "Total Male                              0.0        17352.0   \n",
       "Total                                   2.0        31194.0   \n",
       "\n",
       "                               Number of Homozygotes  Allele Frequency  \n",
       "African                                          0.0          0.000000  \n",
       "African Female                                   0.0          0.000000  \n",
       "African Male                                     0.0          0.000000  \n",
       "Latino                                           0.0          0.000000  \n",
       "Latino Female                                    0.0          0.000000  \n",
       "Latino Male                                      0.0          0.000000  \n",
       "Ashkenazi Jewish                                 0.0          0.000000  \n",
       "Ashkenazi Jewish Female                          0.0          0.000000  \n",
       "Ashkenazi Jewish Male                            0.0          0.000000  \n",
       "East Asian                                       0.0          0.000000  \n",
       "East Asian Female                                0.0          0.000000  \n",
       "East Asian Male                                  0.0          0.000000  \n",
       "European (Finnish)                               0.0          0.000000  \n",
       "European (Finnish) Female                        0.0          0.000000  \n",
       "European (Finnish) Male                          0.0          0.000000  \n",
       "European (non-Finnish)                           0.0          0.000130  \n",
       "European (non-Finnish) Female                    0.0          0.000296  \n",
       "European (non-Finnish) Male                      0.0          0.000000  \n",
       "Other                                            0.0          0.000000  \n",
       "Other Female                                     0.0          0.000000  \n",
       "Other Male                                       0.0          0.000000  \n",
       "South Asian                                      0.0               NaN  \n",
       "South Asian Female                               0.0               NaN  \n",
       "South Asian Male                                 0.0               NaN  \n",
       "Total Female                                     0.0          0.000144  \n",
       "Total Male                                       0.0          0.000000  \n",
       "Total                                            0.0          0.000064  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "build_variant_search_standard_df(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "query = \"\"\"query VariantInRegion($chrom: String!, $start: Int!, $stop: Int!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {\n",
    "  region(start: $start, stop: $stop, chrom: $chrom, reference_genome: $referenceGenome) {\n",
    "    clinvar_variants {\n",
    "      clinical_significance\n",
    "      clinvar_variation_id\n",
    "      gold_stars\n",
    "      major_consequence\n",
    "      pos\n",
    "      variant_id\n",
    "    }\n",
    "    variants(dataset: $datasetId) {\n",
    "      consequence\n",
    "      flags\n",
    "      gene_id\n",
    "      gene_symbol\n",
    "      hgvs\n",
    "      hgvsc\n",
    "      hgvsp\n",
    "      lof\n",
    "      lof_filter\n",
    "      lof_flags\n",
    "      pos\n",
    "      rsid\n",
    "      variant_id: variantId\n",
    "      exome {\n",
    "        ac\n",
    "        ac_hemi\n",
    "        ac_hom\n",
    "        an\n",
    "        af\n",
    "        filters\n",
    "        populations {\n",
    "          id\n",
    "          ac\n",
    "          an\n",
    "          ac_hemi\n",
    "          ac_hom\n",
    "        }\n",
    "      }\n",
    "      genome {\n",
    "        ac\n",
    "        ac_hemi\n",
    "        ac_hom\n",
    "        an\n",
    "        af\n",
    "        filters\n",
    "        populations {\n",
    "          id\n",
    "          ac\n",
    "          an\n",
    "          ac_hemi\n",
    "          ac_hom\n",
    "        }\n",
    "      }\n",
    "    }\n",
    "  }\n",
    "}\n",
    "\"\"\"\n",
    "\n",
    "variables= \"\"\"\n",
    "{\n",
    "  \"datasetId\": \"gnomad_r2_1\",\n",
    "  \"chrom\": \"X\",\n",
    "  \"start\": 980801,\n",
    "  \"stop\": 980830,\n",
    "  \"referenceGenome\": \"GRCh37\"\n",
    "}\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "end_point = \"https://gnomad.broadinstitute.org/api/\"\n",
    "response = requests.post(end_point, data={'query': query, 'variables': variables}, timeout=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = response.json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'region': {'clinvar_variants': [],\n",
       "   'variants': [{'consequence': None,\n",
       "     'flags': ['lcr', 'segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980805,\n",
       "     'rsid': None,\n",
       "     'variant_id': 'X-980805-C-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 1,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 28158,\n",
       "      'af': 3.5513885929398395e-05,\n",
       "      'filters': ['RF'],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 7962,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 812, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 286, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 1528, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 2624, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 1, 'an': 14002, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 944, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['lcr', 'segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980806,\n",
       "     'rsid': None,\n",
       "     'variant_id': 'X-980806-T-C',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 27780,\n",
       "      'af': 7.199424046076313e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 7862,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 766, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 282, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 1528, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 2494, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 2, 'an': 13912, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 936, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['lcr', 'segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980807,\n",
       "     'rsid': None,\n",
       "     'variant_id': 'X-980807-CAA-C',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 3,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 20152,\n",
       "      'af': 0.00014886859865025803,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 6466,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 428, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 214, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 1334, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 1, 'an': 662, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 1, 'an': 10520, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 1, 'an': 528, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['lcr', 'segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980807,\n",
       "     'rsid': None,\n",
       "     'variant_id': 'X-980807-C-CAA',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 11,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 20152,\n",
       "      'af': 0.0005458515283842794,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 10,\n",
       "        'an': 6466,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 1, 'an': 428, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 214, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 1334, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 662, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 10520, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 528, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['lcr', 'segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980807,\n",
       "     'rsid': None,\n",
       "     'variant_id': 'X-980807-C-CA',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 340,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 4,\n",
       "      'an': 20152,\n",
       "      'af': 0.01687177451369591,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 296,\n",
       "        'an': 6466,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 4},\n",
       "       {'id': 'AMR', 'ac': 5, 'an': 428, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 214, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 1, 'an': 1334, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 11, 'an': 662, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 20, 'an': 10520, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 7, 'an': 528, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['lcr', 'segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980807,\n",
       "     'rsid': None,\n",
       "     'variant_id': 'X-980807-C-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 20,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 20152,\n",
       "      'af': 0.0009924573243350536,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 2,\n",
       "        'an': 6466,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 428, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 214, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 1334, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 4, 'an': 662, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 10, 'an': 10520, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 4, 'an': 528, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['lcr', 'segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980807,\n",
       "     'rsid': None,\n",
       "     'variant_id': 'X-980807-CA-C',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 127,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 20152,\n",
       "      'af': 0.006302104009527591,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 15,\n",
       "        'an': 6466,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 10, 'an': 428, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 1, 'an': 214, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 2, 'an': 1334, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 56, 'an': 662, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 37, 'an': 10520, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 6, 'an': 528, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['lcr', 'segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980820,\n",
       "     'rsid': None,\n",
       "     'variant_id': 'X-980820-G-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 29732,\n",
       "      'af': 6.72675904749092e-05,\n",
       "      'filters': ['RF'],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 8380,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 1, 'an': 780, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 280, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 1542, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 3066, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 1, 'an': 14704, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 980, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980828,\n",
       "     'rsid': 'rs201486027',\n",
       "     'variant_id': 'X-980828-ATTT-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 262,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 3,\n",
       "      'an': 31090,\n",
       "      'af': 0.008427146992602124,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 24,\n",
       "        'an': 8650,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 39, 'an': 832, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 3, 'an': 288, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 1, 'an': 1556, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 8, 'an': 3410, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 173, 'an': 15276, 'ac_hemi': 0, 'ac_hom': 2},\n",
       "       {'id': 'OTH', 'ac': 14, 'an': 1078, 'ac_hemi': 0, 'ac_hom': 1},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980828,\n",
       "     'rsid': 'rs201486027',\n",
       "     'variant_id': 'X-980828-ATT-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 1,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 31090,\n",
       "      'af': 3.21646831778707e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 8650,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 1, 'an': 832, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 288, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 1556, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 3410, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 15276, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 1078, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980829,\n",
       "     'rsid': 'rs782304248',\n",
       "     'variant_id': 'X-980829-T-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 38,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 30466,\n",
       "      'af': 0.0012472920632836605,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 37,\n",
       "        'an': 8436,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 1, 'an': 816, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 286, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 1508, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 3316, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 15046, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 1058, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': None,\n",
       "     'flags': ['segdup', 'par'],\n",
       "     'gene_id': None,\n",
       "     'gene_symbol': None,\n",
       "     'hgvs': None,\n",
       "     'hgvsc': None,\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 980830,\n",
       "     'rsid': 'rs781932286',\n",
       "     'variant_id': 'X-980830-T-G',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 1,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 31036,\n",
       "      'af': 3.2220646990591574e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 1,\n",
       "        'an': 8620,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 826, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 286, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 1544, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 3410, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 15270, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 1080, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 0, 'ac_hemi': 0, 'ac_hom': 0}]}}]}}}"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['data']['region']['clinvar_variants']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_df = pd.DataFrame(data['data']['region']['variants'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>exome</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000010404</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*44G&gt;C</td>\n",
       "      <td>c.*44G&gt;C</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>148564233</td>\n",
       "      <td>None</td>\n",
       "      <td>X-148564233-C-G</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 178...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000010404</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*42C&gt;T</td>\n",
       "      <td>c.*42C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>148564235</td>\n",
       "      <td>rs782584433</td>\n",
       "      <td>X-148564235-G-A</td>\n",
       "      <td>{'ac': 0, 'ac_hemi': 0, 'ac_hom': 0, 'an': 178...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000010404</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*37C&gt;T</td>\n",
       "      <td>c.*37C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>148564240</td>\n",
       "      <td>rs192118605</td>\n",
       "      <td>X-148564240-G-A</td>\n",
       "      <td>{'ac': 42, 'ac_hemi': 13, 'ac_hom': 0, 'an': 1...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000010404</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*31G&gt;A</td>\n",
       "      <td>c.*31G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>148564246</td>\n",
       "      <td>rs782432215</td>\n",
       "      <td>X-148564246-C-T</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 2, 'ac_hom': 0, 'an': 181...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000010404</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*29A&gt;G</td>\n",
       "      <td>c.*29A&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>148564248</td>\n",
       "      <td>None</td>\n",
       "      <td>X-148564248-T-C</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 181...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Annotation Flags          gene_id Gene Consequence     hgvsc  \\\n",
       "0  3_prime_UTR_variant    []  ENSG00000010404  IDS    c.*44G>C  c.*44G>C   \n",
       "1  3_prime_UTR_variant    []  ENSG00000010404  IDS    c.*42C>T  c.*42C>T   \n",
       "2  3_prime_UTR_variant    []  ENSG00000010404  IDS    c.*37C>T  c.*37C>T   \n",
       "3  3_prime_UTR_variant    []  ENSG00000010404  IDS    c.*31G>A  c.*31G>A   \n",
       "4  3_prime_UTR_variant    []  ENSG00000010404  IDS    c.*29A>G  c.*29A>G   \n",
       "\n",
       "  hgvsp   lof lof_filter lof_flags        pos         rsid       Variant ID  \\\n",
       "0  None  None       None      None  148564233         None  X-148564233-C-G   \n",
       "1  None  None       None      None  148564235  rs782584433  X-148564235-G-A   \n",
       "2  None  None       None      None  148564240  rs192118605  X-148564240-G-A   \n",
       "3  None  None       None      None  148564246  rs782432215  X-148564246-C-T   \n",
       "4  None  None       None      None  148564248         None  X-148564248-T-C   \n",
       "\n",
       "                                               exome genome  \n",
       "0  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 178...   None  \n",
       "1  {'ac': 0, 'ac_hemi': 0, 'ac_hom': 0, 'an': 178...   None  \n",
       "2  {'ac': 42, 'ac_hemi': 13, 'ac_hom': 0, 'an': 1...   None  \n",
       "3  {'ac': 2, 'ac_hemi': 2, 'ac_hom': 0, 'an': 181...   None  \n",
       "4  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 181...   None  "
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "renamed_cols = {\n",
    "                            'variant_id': 'Variant ID',\n",
    "                            'gene_symbol': 'Gene',\n",
    "                            'hgvs': 'Consequence',\n",
    "                            'consequence': 'Annotation',\n",
    "                            'flags': 'Flags'\n",
    "                        }\n",
    "        \n",
    "standard_cols = [\n",
    "            'Variant ID', 'Gene', 'Consequence', \n",
    "            'Annotation', 'Flags', 'Allele Count',\n",
    "            'Allele Number', 'Allele Frequency',\n",
    "            'Number of Homozygotes'\n",
    "        ]\n",
    "\n",
    "df_renamed = raw_df.rename(columns=renamed_cols)\n",
    "#df_final = self._explicit_allele_informations(df_renamed)\n",
    "#standard_df = df_final.loc[:, standard_cols]\n",
    "#self._add_variant_columns()\n",
    "\n",
    "df_renamed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "def explicit_allele_informations(df):\n",
    "        chromosome = df['Variant ID'][0][0]\n",
    "        allele_count = []\n",
    "        allele_number = []\n",
    "        allele_freq = []\n",
    "        num_homozygotes = []\n",
    "        num_hemizygotes = []\n",
    "        \n",
    "        region = []\n",
    "        \n",
    "        for variant in df['genome']:\n",
    "            if variant:\n",
    "                allele_count.append(variant['ac'])\n",
    "                allele_number.append(variant['an'])\n",
    "                allele_freq.append(variant['af'])\n",
    "                region.append(\"Genome\")\n",
    "            \n",
    "                n_homs = 0\n",
    "                n_hemi = 0\n",
    "                for population in variant['populations']:\n",
    "                    n_homs += population['ac_hom']\n",
    "                    n_hemi += population['ac_hemi']\n",
    "                num_homozygotes.append(n_homs)\n",
    "                num_hemizygotes.append(n_hemi)\n",
    "            else:\n",
    "                region.append(\"Exome\")\n",
    "           \n",
    "        \n",
    "        for variant in df['exome']:\n",
    "            if variant:\n",
    "                allele_count.append(variant['ac'])\n",
    "                allele_number.append(variant['an'])\n",
    "                allele_freq.append(variant['af'])\n",
    "            \n",
    "                n_homs = 0\n",
    "                n_hemi = 0\n",
    "                for population in variant['populations']:\n",
    "                    n_homs += population['ac_hom']\n",
    "                    n_hemi += population['ac_hemi']\n",
    "                num_homozygotes.append(n_homs)\n",
    "                num_hemizygotes.append(n_hemi)\n",
    "                \n",
    "            \n",
    "        df['Allele Count'] = allele_count\n",
    "        df['Allele Number'] = allele_number\n",
    "        df['Allele Frequency'] = allele_freq\n",
    "        df['Number of Homozygotes'] = num_homozygotes\n",
    "        if (chromosome == 'X') or (chromosome == 'Y'):\n",
    "            df['Number of Hemizygotes'] = num_hemizygotes\n",
    "        df['Region'] = region\n",
    "        return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>...</th>\n",
       "      <th>rsid</th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>exome</th>\n",
       "      <th>genome</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Number of Hemizygotes</th>\n",
       "      <th>Region</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000010404</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*44G&gt;C</td>\n",
       "      <td>c.*44G&gt;C</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>X-148564233-C-G</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 178...</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>178182</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Exome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000010404</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*42C&gt;T</td>\n",
       "      <td>c.*42C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>rs782584433</td>\n",
       "      <td>X-148564235-G-A</td>\n",
       "      <td>{'ac': 0, 'ac_hemi': 0, 'ac_hom': 0, 'an': 178...</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>178926</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Exome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000010404</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*37C&gt;T</td>\n",
       "      <td>c.*37C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>rs192118605</td>\n",
       "      <td>X-148564240-G-A</td>\n",
       "      <td>{'ac': 42, 'ac_hemi': 13, 'ac_hom': 0, 'an': 1...</td>\n",
       "      <td>None</td>\n",
       "      <td>42</td>\n",
       "      <td>180127</td>\n",
       "      <td>0.000233</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>Exome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000010404</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*31G&gt;A</td>\n",
       "      <td>c.*31G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>rs782432215</td>\n",
       "      <td>X-148564246-C-T</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 2, 'ac_hom': 0, 'an': 181...</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>181201</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>Exome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000010404</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*29A&gt;G</td>\n",
       "      <td>c.*29A&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>X-148564248-T-C</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 181...</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>181464</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Exome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Annotation Flags          gene_id Gene Consequence     hgvsc  \\\n",
       "0  3_prime_UTR_variant    []  ENSG00000010404  IDS    c.*44G>C  c.*44G>C   \n",
       "1  3_prime_UTR_variant    []  ENSG00000010404  IDS    c.*42C>T  c.*42C>T   \n",
       "2  3_prime_UTR_variant    []  ENSG00000010404  IDS    c.*37C>T  c.*37C>T   \n",
       "3  3_prime_UTR_variant    []  ENSG00000010404  IDS    c.*31G>A  c.*31G>A   \n",
       "4  3_prime_UTR_variant    []  ENSG00000010404  IDS    c.*29A>G  c.*29A>G   \n",
       "\n",
       "  hgvsp   lof lof_filter lof_flags  ...         rsid       Variant ID  \\\n",
       "0  None  None       None      None  ...         None  X-148564233-C-G   \n",
       "1  None  None       None      None  ...  rs782584433  X-148564235-G-A   \n",
       "2  None  None       None      None  ...  rs192118605  X-148564240-G-A   \n",
       "3  None  None       None      None  ...  rs782432215  X-148564246-C-T   \n",
       "4  None  None       None      None  ...         None  X-148564248-T-C   \n",
       "\n",
       "                                               exome genome Allele Count  \\\n",
       "0  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 178...   None            1   \n",
       "1  {'ac': 0, 'ac_hemi': 0, 'ac_hom': 0, 'an': 178...   None            0   \n",
       "2  {'ac': 42, 'ac_hemi': 13, 'ac_hom': 0, 'an': 1...   None           42   \n",
       "3  {'ac': 2, 'ac_hemi': 2, 'ac_hom': 0, 'an': 181...   None            2   \n",
       "4  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 181...   None            1   \n",
       "\n",
       "   Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0         178182          0.000006                      0   \n",
       "1         178926          0.000000                      0   \n",
       "2         180127          0.000233                      0   \n",
       "3         181201          0.000011                      0   \n",
       "4         181464          0.000006                      0   \n",
       "\n",
       "   Number of Hemizygotes  Region  \n",
       "0                      0   Exome  \n",
       "1                      0   Exome  \n",
       "2                     13   Exome  \n",
       "3                      2   Exome  \n",
       "4                      0   Exome  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = explicit_allele_informations(df_renamed)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "standard_cols = [\n",
    "                    'Variant ID', 'Gene', 'Consequence', \n",
    "                    'Annotation', 'Flags', 'Allele Count',\n",
    "                    'Allele Number', 'Allele Frequency',\n",
    "                    'Number of Homozygotes', 'Number of Hemizygotes',\n",
    "                    'Region'\n",
    "                ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Number of Hemizygotes</th>\n",
       "      <th>Region</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>X-148564233-C-G</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*44G&gt;C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>178182</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Exome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>X-148564235-G-A</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*42C&gt;T</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>178926</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Exome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>X-148564240-G-A</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*37C&gt;T</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>42</td>\n",
       "      <td>180127</td>\n",
       "      <td>0.000233</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>Exome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>X-148564246-C-T</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*31G&gt;A</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>181201</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>Exome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>X-148564248-T-C</td>\n",
       "      <td>IDS</td>\n",
       "      <td>c.*29A&gt;G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>181464</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Exome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Variant ID Gene Consequence           Annotation Flags  Allele Count  \\\n",
       "0  X-148564233-C-G  IDS    c.*44G>C  3_prime_UTR_variant    []             1   \n",
       "1  X-148564235-G-A  IDS    c.*42C>T  3_prime_UTR_variant    []             0   \n",
       "2  X-148564240-G-A  IDS    c.*37C>T  3_prime_UTR_variant    []            42   \n",
       "3  X-148564246-C-T  IDS    c.*31G>A  3_prime_UTR_variant    []             2   \n",
       "4  X-148564248-T-C  IDS    c.*29A>G  3_prime_UTR_variant    []             1   \n",
       "\n",
       "   Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0         178182          0.000006                      0   \n",
       "1         178926          0.000000                      0   \n",
       "2         180127          0.000233                      0   \n",
       "3         181201          0.000011                      0   \n",
       "4         181464          0.000006                      0   \n",
       "\n",
       "   Number of Hemizygotes Region  \n",
       "0                      0  Exome  \n",
       "1                      0  Exome  \n",
       "2                     13  Exome  \n",
       "3                      2  Exome  \n",
       "4                      0  Exome  "
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[:, standard_cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'region': {'clinvar_variants': [],\n",
       "   'variants': [{'consequence': '3_prime_UTR_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000010404',\n",
       "     'gene_symbol': 'IDS',\n",
       "     'hgvs': 'c.*44G>C',\n",
       "     'hgvsc': 'c.*44G>C',\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 148564233,\n",
       "     'rsid': None,\n",
       "     'variant_id': 'X-148564233-C-G',\n",
       "     'exome': {'ac': 1,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 178182,\n",
       "      'af': 5.6122391711845195e-06,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 12922,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 27202, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 7416, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 13658, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 14354, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 1, 'an': 79541, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 4441, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 18648, 'ac_hemi': 0, 'ac_hom': 0}]},\n",
       "     'genome': None},\n",
       "    {'consequence': '3_prime_UTR_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000010404',\n",
       "     'gene_symbol': 'IDS',\n",
       "     'hgvs': 'c.*42C>T',\n",
       "     'hgvsc': 'c.*42C>T',\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 148564235,\n",
       "     'rsid': 'rs782584433',\n",
       "     'variant_id': 'X-148564235-G-A',\n",
       "     'exome': {'ac': 0,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 178926,\n",
       "      'af': 0,\n",
       "      'filters': ['AC0'],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 12963,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 27247, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 7423, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 13705, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 14519, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 79888, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 4459, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 18722, 'ac_hemi': 0, 'ac_hom': 0}]},\n",
       "     'genome': None},\n",
       "    {'consequence': '3_prime_UTR_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000010404',\n",
       "     'gene_symbol': 'IDS',\n",
       "     'hgvs': 'c.*37C>T',\n",
       "     'hgvsc': 'c.*37C>T',\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 148564240,\n",
       "     'rsid': 'rs192118605',\n",
       "     'variant_id': 'X-148564240-G-A',\n",
       "     'exome': {'ac': 42,\n",
       "      'ac_hemi': 13,\n",
       "      'ac_hom': 0,\n",
       "      'an': 180127,\n",
       "      'af': 0.00023316881977715722,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 13026,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 27290, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 7445, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 41, 'an': 13754, 'ac_hemi': 13, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 14786, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 1, 'an': 80524, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 4479, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 18823, 'ac_hemi': 0, 'ac_hom': 0}]},\n",
       "     'genome': None},\n",
       "    {'consequence': '3_prime_UTR_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000010404',\n",
       "     'gene_symbol': 'IDS',\n",
       "     'hgvs': 'c.*31G>A',\n",
       "     'hgvsc': 'c.*31G>A',\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 148564246,\n",
       "     'rsid': 'rs782432215',\n",
       "     'variant_id': 'X-148564246-C-T',\n",
       "     'exome': {'ac': 2,\n",
       "      'ac_hemi': 2,\n",
       "      'ac_hom': 0,\n",
       "      'an': 181201,\n",
       "      'af': 1.1037466680647458e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 13080,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 27353, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 7454, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 13822, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 15112, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 80971, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 4501, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 2, 'an': 18908, 'ac_hemi': 2, 'ac_hom': 0}]},\n",
       "     'genome': None},\n",
       "    {'consequence': '3_prime_UTR_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000010404',\n",
       "     'gene_symbol': 'IDS',\n",
       "     'hgvs': 'c.*29A>G',\n",
       "     'hgvsc': 'c.*29A>G',\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 148564248,\n",
       "     'rsid': None,\n",
       "     'variant_id': 'X-148564248-T-C',\n",
       "     'exome': {'ac': 1,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 181464,\n",
       "      'af': 5.510734911607812e-06,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 13090,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 1, 'an': 27361, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 7466, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 13827, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 15187, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 81104, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 4506, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 18923, 'ac_hemi': 0, 'ac_hom': 0}]},\n",
       "     'genome': None}]}}}"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
