{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Region Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Trp402Ter</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>[]</td>\n",
       "      <td>120</td>\n",
       "      <td>142464</td>\n",
       "      <td>0.000842</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-1002750-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>141870</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002750</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-1002751-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143004</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002751</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-1002756-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val405Gly</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>141746</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002756</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-1002759-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser406Trp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>142602</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002759</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4-1002766-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala408Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>143012</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002766</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4-1002767-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Arg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2750</td>\n",
       "      <td>142238</td>\n",
       "      <td>0.019334</td>\n",
       "      <td>67</td>\n",
       "      <td>4</td>\n",
       "      <td>1002767</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-1002768-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Glu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143002</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4-1002770-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Thr410Pro</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>140858</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4-1002770-AC-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val411SerfsTer29</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>140860</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant ID  Gene         Consequence          Annotation Flags  \\\n",
       "0   4-1002747-G-A  IDUA         p.Trp402Ter         stop_gained    []   \n",
       "1   4-1002750-C-T  IDUA         p.Ala403Val    missense_variant    []   \n",
       "2   4-1002751-C-G  IDUA         p.Ala403Ala  synonymous_variant    []   \n",
       "3   4-1002756-T-G  IDUA         p.Val405Gly    missense_variant    []   \n",
       "4   4-1002759-C-G  IDUA         p.Ser406Trp    missense_variant    []   \n",
       "5   4-1002766-C-T  IDUA         p.Ala408Ala  synonymous_variant    []   \n",
       "6   4-1002767-G-C  IDUA         p.Gly409Arg    missense_variant    []   \n",
       "7   4-1002768-G-A  IDUA         p.Gly409Glu    missense_variant    []   \n",
       "8   4-1002770-A-C  IDUA         p.Thr410Pro    missense_variant    []   \n",
       "9  4-1002770-AC-A  IDUA  p.Val411SerfsTer29  frameshift_variant    []   \n",
       "\n",
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0           120         142464          0.000842                      0   \n",
       "1             2         141870          0.000014                      0   \n",
       "2             2         143004          0.000014                      0   \n",
       "3             1         141746          0.000007                      0   \n",
       "4             2         142602          0.000014                      0   \n",
       "5             3         143012          0.000021                      0   \n",
       "6          2750         142238          0.019334                     67   \n",
       "7             2         143002          0.000014                      0   \n",
       "8             5         140858          0.000035                      0   \n",
       "9             1         140860          0.000007                      0   \n",
       "\n",
       "  Chromosome Location Reference Alternative  \n",
       "0          4  1002747         G           A  \n",
       "1          4  1002750         C           T  \n",
       "2          4  1002751         C           G  \n",
       "3          4  1002756         T           G  \n",
       "4          4  1002759         C           G  \n",
       "5          4  1002766         C           T  \n",
       "6          4  1002767         G           C  \n",
       "7          4  1002768         G           A  \n",
       "8          4  1002770         A           C  \n",
       "9          4  1002770        AC           A  "
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import RegionSearch\n",
    "rs = RegionSearch(3, 4, 1002741, 1002771)\n",
    "normal, clinical = rs.get_data()\n",
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>clinvar_variation_id</th>\n",
       "      <th>gold_stars</th>\n",
       "      <th>major_consequence</th>\n",
       "      <th>pos</th>\n",
       "      <th>variant_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[Pathogenic]</td>\n",
       "      <td>11908</td>\n",
       "      <td>2</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1002747</td>\n",
       "      <td>4-1002747-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[Pathogenic]</td>\n",
       "      <td>555230</td>\n",
       "      <td>1</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1002748</td>\n",
       "      <td>4-1002748-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[Pathogenic]</td>\n",
       "      <td>552095</td>\n",
       "      <td>2</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1002752</td>\n",
       "      <td>4-1002752-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[Likely_benign]</td>\n",
       "      <td>724507</td>\n",
       "      <td>1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1002766</td>\n",
       "      <td>4-1002766-C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[Conflicting_interpretations_of_pathogenicity]</td>\n",
       "      <td>92627</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1002767</td>\n",
       "      <td>4-1002767-G-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>[Uncertain_significance]</td>\n",
       "      <td>557616</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1002768</td>\n",
       "      <td>4-1002768-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>[Uncertain_significance]</td>\n",
       "      <td>593572</td>\n",
       "      <td>1</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>1002771</td>\n",
       "      <td>4-1002771-CCGTCCTGGACAGCAACCACACGGTGGG-C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            clinical_significance clinvar_variation_id  \\\n",
       "0                                    [Pathogenic]                11908   \n",
       "1                                    [Pathogenic]               555230   \n",
       "2                                    [Pathogenic]               552095   \n",
       "3                                 [Likely_benign]               724507   \n",
       "4  [Conflicting_interpretations_of_pathogenicity]                92627   \n",
       "5                        [Uncertain_significance]               557616   \n",
       "6                        [Uncertain_significance]               593572   \n",
       "\n",
       "   gold_stars   major_consequence      pos  \\\n",
       "0           2         stop_gained  1002747   \n",
       "1           1         stop_gained  1002748   \n",
       "2           2         stop_gained  1002752   \n",
       "3           1  synonymous_variant  1002766   \n",
       "4           1    missense_variant  1002767   \n",
       "5           1    missense_variant  1002768   \n",
       "6           1    inframe_deletion  1002771   \n",
       "\n",
       "                                 variant_id  \n",
       "0                             4-1002747-G-A  \n",
       "1                             4-1002748-G-A  \n",
       "2                             4-1002752-G-T  \n",
       "3                             4-1002766-C-T  \n",
       "4                             4-1002767-G-C  \n",
       "5                             4-1002768-G-A  \n",
       "6  4-1002771-CCGTCCTGGACAGCAACCACACGGTGGG-C  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>...</th>\n",
       "      <th>Alternative</th>\n",
       "      <th>African</th>\n",
       "      <th>Amish</th>\n",
       "      <th>Latino</th>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <th>East Asian</th>\n",
       "      <th>European (Finnish)</th>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <th>Other</th>\n",
       "      <th>South Asian</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Trp402Ter</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>[]</td>\n",
       "      <td>120</td>\n",
       "      <td>142464</td>\n",
       "      <td>0.000842</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>A</td>\n",
       "      <td>3.122898e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.205882e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.906413e-04</td>\n",
       "      <td>1.552843e-03</td>\n",
       "      <td>4.668534e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-1002750-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>141870</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>T</td>\n",
       "      <td>4.811856e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-1002751-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143004</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.103181e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-1002756-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val405Gly</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>141746</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.561378e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-1002759-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser406Trp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>142602</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.108873e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4-1002766-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala408Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>143012</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.656072e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4-1002767-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Arg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2750</td>\n",
       "      <td>142238</td>\n",
       "      <td>0.019334</td>\n",
       "      <td>67</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>5.884051e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.006465e-02</td>\n",
       "      <td>9.052505e-03</td>\n",
       "      <td>1.856594e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.666791e-04</td>\n",
       "      <td>2.192164e-02</td>\n",
       "      <td>6.609385e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-1002768-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Glu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143002</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.104144e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4-1002770-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Thr410Pro</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>140858</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>1.217552e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4-1002770-AC-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val411SerfsTer29</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>140860</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>A</td>\n",
       "      <td>2.435104e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant ID  Gene         Consequence          Annotation Flags  \\\n",
       "0   4-1002747-G-A  IDUA         p.Trp402Ter         stop_gained    []   \n",
       "1   4-1002750-C-T  IDUA         p.Ala403Val    missense_variant    []   \n",
       "2   4-1002751-C-G  IDUA         p.Ala403Ala  synonymous_variant    []   \n",
       "3   4-1002756-T-G  IDUA         p.Val405Gly    missense_variant    []   \n",
       "4   4-1002759-C-G  IDUA         p.Ser406Trp    missense_variant    []   \n",
       "5   4-1002766-C-T  IDUA         p.Ala408Ala  synonymous_variant    []   \n",
       "6   4-1002767-G-C  IDUA         p.Gly409Arg    missense_variant    []   \n",
       "7   4-1002768-G-A  IDUA         p.Gly409Glu    missense_variant    []   \n",
       "8   4-1002770-A-C  IDUA         p.Thr410Pro    missense_variant    []   \n",
       "9  4-1002770-AC-A  IDUA  p.Val411SerfsTer29  frameshift_variant    []   \n",
       "\n",
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0           120         142464          0.000842                      0   \n",
       "1             2         141870          0.000014                      0   \n",
       "2             2         143004          0.000014                      0   \n",
       "3             1         141746          0.000007                      0   \n",
       "4             2         142602          0.000014                      0   \n",
       "5             3         143012          0.000021                      0   \n",
       "6          2750         142238          0.019334                     67   \n",
       "7             2         143002          0.000014                      0   \n",
       "8             5         140858          0.000035                      0   \n",
       "9             1         140860          0.000007                      0   \n",
       "\n",
       "  Chromosome  ... Alternative       African         Amish        Latino  \\\n",
       "0          4  ...           A  3.122898e-04  0.000000e+00  2.205882e-04   \n",
       "1          4  ...           T  4.811856e-05  0.000000e+00  0.000000e+00   \n",
       "2          4  ...           G  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "3          4  ...           G  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "4          4  ...           G  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "5          4  ...           T  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "6          4  ...           C  5.884051e-02  0.000000e+00  1.006465e-02   \n",
       "7          4  ...           A  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "8          4  ...           C  1.217552e-04  0.000000e+00  0.000000e+00   \n",
       "9          4  ...           A  2.435104e-05  0.000000e+00  0.000000e+00   \n",
       "\n",
       "  Ashkenazi Jewish    East Asian European (Finnish) European (non-Finnish)  \\\n",
       "0     0.000000e+00  0.000000e+00       2.906413e-04           1.552843e-03   \n",
       "1     0.000000e+00  0.000000e+00       0.000000e+00           0.000000e+00   \n",
       "2     0.000000e+00  0.000000e+00       0.000000e+00           3.103181e-05   \n",
       "3     0.000000e+00  0.000000e+00       0.000000e+00           1.561378e-05   \n",
       "4     0.000000e+00  0.000000e+00       0.000000e+00           3.108873e-05   \n",
       "5     0.000000e+00  0.000000e+00       0.000000e+00           4.656072e-05   \n",
       "6     9.052505e-03  1.856594e-02       0.000000e+00           4.666791e-04   \n",
       "7     0.000000e+00  0.000000e+00       0.000000e+00           3.104144e-05   \n",
       "8     0.000000e+00  0.000000e+00       0.000000e+00           0.000000e+00   \n",
       "9     0.000000e+00  0.000000e+00       0.000000e+00           0.000000e+00   \n",
       "\n",
       "          Other   South Asian  \n",
       "0  4.668534e-04  0.000000e+00  \n",
       "1  0.000000e+00  0.000000e+00  \n",
       "2  0.000000e+00  0.000000e+00  \n",
       "3  0.000000e+00  0.000000e+00  \n",
       "4  0.000000e+00  0.000000e+00  \n",
       "5  0.000000e+00  0.000000e+00  \n",
       "6  2.192164e-02  6.609385e-04  \n",
       "7  0.000000e+00  0.000000e+00  \n",
       "8  0.000000e+00  0.000000e+00  \n",
       "9  0.000000e+00  0.000000e+00  \n",
       "\n",
       "[10 rows x 22 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal, _ = rs.get_data(additional_population_info=True)\n",
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "      <th>African</th>\n",
       "      <th>Amish</th>\n",
       "      <th>Latino</th>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <th>East Asian</th>\n",
       "      <th>European (Finnish)</th>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <th>Other</th>\n",
       "      <th>South Asian</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>120</td>\n",
       "      <td>142464</td>\n",
       "      <td>0.000842</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3.122898e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.205882e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.906413e-04</td>\n",
       "      <td>1.552843e-03</td>\n",
       "      <td>4.668534e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>141870</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002750</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4.811856e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>143004</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002751</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.103181e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>141746</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002756</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.561378e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>142602</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002759</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.108873e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3</td>\n",
       "      <td>143012</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002766</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.656072e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2750</td>\n",
       "      <td>142238</td>\n",
       "      <td>0.019334</td>\n",
       "      <td>67</td>\n",
       "      <td>4</td>\n",
       "      <td>1002767</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5.884051e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.006465e-02</td>\n",
       "      <td>9.052505e-03</td>\n",
       "      <td>1.856594e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.666791e-04</td>\n",
       "      <td>2.192164e-02</td>\n",
       "      <td>6.609385e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>143002</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.104144e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5</td>\n",
       "      <td>140858</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1.217552e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>140860</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>2.435104e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0           120         142464          0.000842                      0   \n",
       "1             2         141870          0.000014                      0   \n",
       "2             2         143004          0.000014                      0   \n",
       "3             1         141746          0.000007                      0   \n",
       "4             2         142602          0.000014                      0   \n",
       "5             3         143012          0.000021                      0   \n",
       "6          2750         142238          0.019334                     67   \n",
       "7             2         143002          0.000014                      0   \n",
       "8             5         140858          0.000035                      0   \n",
       "9             1         140860          0.000007                      0   \n",
       "\n",
       "  Chromosome Location Reference Alternative       African         Amish  \\\n",
       "0          4  1002747         G           A  3.122898e-04  0.000000e+00   \n",
       "1          4  1002750         C           T  4.811856e-05  0.000000e+00   \n",
       "2          4  1002751         C           G  0.000000e+00  0.000000e+00   \n",
       "3          4  1002756         T           G  0.000000e+00  0.000000e+00   \n",
       "4          4  1002759         C           G  0.000000e+00  0.000000e+00   \n",
       "5          4  1002766         C           T  0.000000e+00  0.000000e+00   \n",
       "6          4  1002767         G           C  5.884051e-02  0.000000e+00   \n",
       "7          4  1002768         G           A  0.000000e+00  0.000000e+00   \n",
       "8          4  1002770         A           C  1.217552e-04  0.000000e+00   \n",
       "9          4  1002770        AC           A  2.435104e-05  0.000000e+00   \n",
       "\n",
       "         Latino Ashkenazi Jewish    East Asian European (Finnish)  \\\n",
       "0  2.205882e-04     0.000000e+00  0.000000e+00       2.906413e-04   \n",
       "1  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "2  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "3  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "4  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "5  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "6  1.006465e-02     9.052505e-03  1.856594e-02       0.000000e+00   \n",
       "7  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "8  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "9  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "\n",
       "  European (non-Finnish)         Other   South Asian  \n",
       "0           1.552843e-03  4.668534e-04  0.000000e+00  \n",
       "1           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "2           3.103181e-05  0.000000e+00  0.000000e+00  \n",
       "3           1.561378e-05  0.000000e+00  0.000000e+00  \n",
       "4           3.108873e-05  0.000000e+00  0.000000e+00  \n",
       "5           4.656072e-05  0.000000e+00  0.000000e+00  \n",
       "6           4.666791e-04  2.192164e-02  6.609385e-04  \n",
       "7           3.104144e-05  0.000000e+00  0.000000e+00  \n",
       "8           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "9           0.000000e+00  0.000000e+00  0.000000e+00  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal.iloc[:, 5:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Number of Hemizygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>X-1002741-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>1</td>\n",
       "      <td>31378</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002741</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>X-1002742-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>2</td>\n",
       "      <td>31316</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002742</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>X-1002746-TATCA-T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>1</td>\n",
       "      <td>31376</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002746</td>\n",
       "      <td>TATCA</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>X-1002765-TTATC-T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>9</td>\n",
       "      <td>31374</td>\n",
       "      <td>0.000287</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002765</td>\n",
       "      <td>TTATC</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>X-1002767-ATCTATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>1</td>\n",
       "      <td>31348</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002767</td>\n",
       "      <td>ATCTATCTT</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>X-1002771-ATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>2</td>\n",
       "      <td>31338</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002771</td>\n",
       "      <td>ATCTT</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Variant ID  Gene Consequence Annotation          Flags  \\\n",
       "0          X-1002741-C-T  None        None       None  [segdup, par]   \n",
       "1          X-1002742-G-A  None        None       None  [segdup, par]   \n",
       "2      X-1002746-TATCA-T  None        None       None  [segdup, par]   \n",
       "3      X-1002765-TTATC-T  None        None       None  [segdup, par]   \n",
       "4  X-1002767-ATCTATCTT-A  None        None       None  [segdup, par]   \n",
       "5      X-1002771-ATCTT-A  None        None       None  [segdup, par]   \n",
       "\n",
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0             1          31378          0.000032                      0   \n",
       "1             2          31316          0.000064                      0   \n",
       "2             1          31376          0.000032                      0   \n",
       "3             9          31374          0.000287                      0   \n",
       "4             1          31348          0.000032                      0   \n",
       "5             2          31338          0.000064                      0   \n",
       "\n",
       "   Number of Hemizygotes  Source Chromosome Location  Reference Alternative  \n",
       "0                      0  Genome          X  1002741          C           T  \n",
       "1                      0  Genome          X  1002742          G           A  \n",
       "2                      0  Genome          X  1002746      TATCA           T  \n",
       "3                      0  Genome          X  1002765      TTATC           T  \n",
       "4                      0  Genome          X  1002767  ATCTATCTT           A  \n",
       "5                      0  Genome          X  1002771      ATCTT           A  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs = RegionSearch(2, 'X', 1002741, 1002771)\n",
    "normal, clinical = rs.get_data()\n",
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>exome</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002741</td>\n",
       "      <td>rs782385877</td>\n",
       "      <td>X-1002741-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002742</td>\n",
       "      <td>rs181914603</td>\n",
       "      <td>X-1002742-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002746</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002746-TATCA-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002765</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002765-TTATC-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 9, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002767</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002767-ATCTATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>None</td>\n",
       "      <td>[segdup, par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002771</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002771-ATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  consequence          flags gene_id gene_symbol  hgvs hgvsc hgvsp   lof  \\\n",
       "0        None  [segdup, par]    None        None  None  None  None  None   \n",
       "1        None  [segdup, par]    None        None  None  None  None  None   \n",
       "2        None  [segdup, par]    None        None  None  None  None  None   \n",
       "3        None  [segdup, par]    None        None  None  None  None  None   \n",
       "4        None  [segdup, par]    None        None  None  None  None  None   \n",
       "5        None  [segdup, par]    None        None  None  None  None  None   \n",
       "\n",
       "  lof_filter lof_flags      pos         rsid             variant_id exome  \\\n",
       "0       None      None  1002741  rs782385877          X-1002741-C-T  None   \n",
       "1       None      None  1002742  rs181914603          X-1002742-G-A  None   \n",
       "2       None      None  1002746         None      X-1002746-TATCA-T  None   \n",
       "3       None      None  1002765         None      X-1002765-TTATC-T  None   \n",
       "4       None      None  1002767         None  X-1002767-ATCTATCTT-A  None   \n",
       "5       None      None  1002771         None      X-1002771-ATCTT-A  None   \n",
       "\n",
       "                                              genome  \n",
       "0  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "1  {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "2  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "3  {'ac': 9, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "4  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "5  {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal, _ = rs.get_data(standard=False)\n",
    "normal"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-987000-A-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-85A&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>140350</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>987000</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-987002-A-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-83A&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>140342</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>987002</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-987003-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-82T&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>139354</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>987003</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-987004-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-81G&gt;A</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>142964</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>987004</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-987005-G-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-80G&gt;T</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>143228</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>987005</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5525</th>\n",
       "      <td>4-1004487-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*94C&gt;T</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>4</td>\n",
       "      <td>143296</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1004487</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5526</th>\n",
       "      <td>4-1004491-A-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*98A&gt;T</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>143074</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1004491</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5527</th>\n",
       "      <td>4-1004492-T-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*99T&gt;C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143280</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1004492</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5528</th>\n",
       "      <td>4-1004503-TTTA-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>None</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>25</td>\n",
       "      <td>143260</td>\n",
       "      <td>0.000175</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1004503</td>\n",
       "      <td>TTTA</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5529</th>\n",
       "      <td>4-1004505-T-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*112T&gt;C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>143248</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1004505</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5530 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Variant ID  Gene Consequence           Annotation Flags  \\\n",
       "0         4-987000-A-G  IDUA    c.-85A>G  5_prime_UTR_variant    []   \n",
       "1         4-987002-A-G  IDUA    c.-83A>G  5_prime_UTR_variant    []   \n",
       "2         4-987003-T-G  IDUA    c.-82T>G  5_prime_UTR_variant    []   \n",
       "3         4-987004-G-A  IDUA    c.-81G>A  5_prime_UTR_variant    []   \n",
       "4         4-987005-G-T  IDUA    c.-80G>T  5_prime_UTR_variant    []   \n",
       "...                ...   ...         ...                  ...   ...   \n",
       "5525     4-1004487-C-T  IDUA    c.*94C>T  3_prime_UTR_variant    []   \n",
       "5526     4-1004491-A-T  IDUA    c.*98A>T  3_prime_UTR_variant    []   \n",
       "5527     4-1004492-T-C  IDUA    c.*99T>C  3_prime_UTR_variant    []   \n",
       "5528  4-1004503-TTTA-T  IDUA        None  3_prime_UTR_variant    []   \n",
       "5529     4-1004505-T-C  IDUA   c.*112T>C  3_prime_UTR_variant    []   \n",
       "\n",
       "      Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0                1         140350          0.000007                      0   \n",
       "1                1         140342          0.000007                      0   \n",
       "2                1         139354          0.000007                      0   \n",
       "3                1         142964          0.000007                      0   \n",
       "4                1         143228          0.000007                      0   \n",
       "...            ...            ...               ...                    ...   \n",
       "5525             4         143296          0.000028                      0   \n",
       "5526             1         143074          0.000007                      0   \n",
       "5527             2         143280          0.000014                      0   \n",
       "5528            25         143260          0.000175                      0   \n",
       "5529             1         143248          0.000007                      0   \n",
       "\n",
       "     Chromosome Location Reference Alternative  \n",
       "0             4   987000         A           G  \n",
       "1             4   987002         A           G  \n",
       "2             4   987003         T           G  \n",
       "3             4   987004         G           A  \n",
       "4             4   987005         G           T  \n",
       "...         ...      ...       ...         ...  \n",
       "5525          4  1004487         C           T  \n",
       "5526          4  1004491         A           T  \n",
       "5527          4  1004492         T           C  \n",
       "5528          4  1004503      TTTA           T  \n",
       "5529          4  1004505         T           C  \n",
       "\n",
       "[5530 rows x 13 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import GeneSearch\n",
    "gs = GeneSearch(3, \"idua\")\n",
    "normal, _ = gs.get_data()\n",
    "normal"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Variant Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Allele Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>African</th>\n",
       "      <td>13</td>\n",
       "      <td>41628</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African Female</th>\n",
       "      <td>6</td>\n",
       "      <td>22480</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000267</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African Male</th>\n",
       "      <td>7</td>\n",
       "      <td>19148</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amish</th>\n",
       "      <td>0</td>\n",
       "      <td>898</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amish Female</th>\n",
       "      <td>0</td>\n",
       "      <td>470</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amish Male</th>\n",
       "      <td>0</td>\n",
       "      <td>428</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino</th>\n",
       "      <td>3</td>\n",
       "      <td>13600</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino Female</th>\n",
       "      <td>2</td>\n",
       "      <td>5882</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino Male</th>\n",
       "      <td>1</td>\n",
       "      <td>7718</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <td>0</td>\n",
       "      <td>3316</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish Female</th>\n",
       "      <td>0</td>\n",
       "      <td>1756</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish Male</th>\n",
       "      <td>0</td>\n",
       "      <td>1560</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian</th>\n",
       "      <td>0</td>\n",
       "      <td>3124</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Female</th>\n",
       "      <td>0</td>\n",
       "      <td>1440</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Male</th>\n",
       "      <td>0</td>\n",
       "      <td>1684</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish)</th>\n",
       "      <td>3</td>\n",
       "      <td>10322</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) Female</th>\n",
       "      <td>1</td>\n",
       "      <td>2476</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) Male</th>\n",
       "      <td>2</td>\n",
       "      <td>7846</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <td>100</td>\n",
       "      <td>64398</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Female</th>\n",
       "      <td>59</td>\n",
       "      <td>37292</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Male</th>\n",
       "      <td>41</td>\n",
       "      <td>27106</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>1</td>\n",
       "      <td>2142</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other Female</th>\n",
       "      <td>1</td>\n",
       "      <td>1094</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other Male</th>\n",
       "      <td>0</td>\n",
       "      <td>1048</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian</th>\n",
       "      <td>0</td>\n",
       "      <td>3036</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian Female</th>\n",
       "      <td>0</td>\n",
       "      <td>560</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian Male</th>\n",
       "      <td>0</td>\n",
       "      <td>2476</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total Female</th>\n",
       "      <td>69</td>\n",
       "      <td>73450</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total Male</th>\n",
       "      <td>51</td>\n",
       "      <td>69014</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000739</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>120</td>\n",
       "      <td>142464</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000842</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Allele Count  Allele Number  \\\n",
       "African                                  13          41628   \n",
       "African Female                            6          22480   \n",
       "African Male                              7          19148   \n",
       "Amish                                     0            898   \n",
       "Amish Female                              0            470   \n",
       "Amish Male                                0            428   \n",
       "Latino                                    3          13600   \n",
       "Latino Female                             2           5882   \n",
       "Latino Male                               1           7718   \n",
       "Ashkenazi Jewish                          0           3316   \n",
       "Ashkenazi Jewish Female                   0           1756   \n",
       "Ashkenazi Jewish Male                     0           1560   \n",
       "East Asian                                0           3124   \n",
       "East Asian Female                         0           1440   \n",
       "East Asian Male                           0           1684   \n",
       "European (Finnish)                        3          10322   \n",
       "European (Finnish) Female                 1           2476   \n",
       "European (Finnish) Male                   2           7846   \n",
       "European (non-Finnish)                  100          64398   \n",
       "European (non-Finnish) Female            59          37292   \n",
       "European (non-Finnish) Male              41          27106   \n",
       "Other                                     1           2142   \n",
       "Other Female                              1           1094   \n",
       "Other Male                                0           1048   \n",
       "South Asian                               0           3036   \n",
       "South Asian Female                        0            560   \n",
       "South Asian Male                          0           2476   \n",
       "Total Female                             69          73450   \n",
       "Total Male                               51          69014   \n",
       "Total                                   120         142464   \n",
       "\n",
       "                               Number of Homozygotes  Allele Frequency  \n",
       "African                                            0          0.000312  \n",
       "African Female                                     0          0.000267  \n",
       "African Male                                       0          0.000366  \n",
       "Amish                                              0          0.000000  \n",
       "Amish Female                                       0          0.000000  \n",
       "Amish Male                                         0          0.000000  \n",
       "Latino                                             0          0.000221  \n",
       "Latino Female                                      0          0.000340  \n",
       "Latino Male                                        0          0.000130  \n",
       "Ashkenazi Jewish                                   0          0.000000  \n",
       "Ashkenazi Jewish Female                            0          0.000000  \n",
       "Ashkenazi Jewish Male                              0          0.000000  \n",
       "East Asian                                         0          0.000000  \n",
       "East Asian Female                                  0          0.000000  \n",
       "East Asian Male                                    0          0.000000  \n",
       "European (Finnish)                                 0          0.000291  \n",
       "European (Finnish) Female                          0          0.000404  \n",
       "European (Finnish) Male                            0          0.000255  \n",
       "European (non-Finnish)                             0          0.001553  \n",
       "European (non-Finnish) Female                      0          0.001582  \n",
       "European (non-Finnish) Male                        0          0.001513  \n",
       "Other                                              0          0.000467  \n",
       "Other Female                                       0          0.000914  \n",
       "Other Male                                         0          0.000000  \n",
       "South Asian                                        0          0.000000  \n",
       "South Asian Female                                 0          0.000000  \n",
       "South Asian Male                                   0          0.000000  \n",
       "Total Female                                       0          0.000939  \n",
       "Total Male                                         0          0.000739  \n",
       "Total                                              0          0.000842  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import VariantSearch\n",
    "vs = VariantSearch(3, '4-1002747-G-A')\n",
    "df, meta = vs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'variantId': '4-1002747-G-A',\n",
       " 'reference_genome': 'GRCh38',\n",
       " 'chrom': '4',\n",
       " 'pos': 1002747,\n",
       " 'ref': 'G',\n",
       " 'alt': 'A',\n",
       " 'colocatedVariants': ['4-1002747-G-T'],\n",
       " 'multiNucleotideVariants': [],\n",
       " 'exome': None,\n",
       " 'genome': {'ac': 120,\n",
       "  'an': 142464,\n",
       "  'ac_hemi': 0,\n",
       "  'ac_hom': 0,\n",
       "  'faf95': {'popmax': 0.0013063199999999997, 'popmax_population': 'NFE'},\n",
       "  'filters': [],\n",
       "  'age_distribution': None,\n",
       "  'qualityMetrics': {'alleleBalance': {'alt': {'bin_edges': [0,\n",
       "      0.05,\n",
       "      0.1,\n",
       "      0.15,\n",
       "      0.2,\n",
       "      0.25,\n",
       "      0.3,\n",
       "      0.35,\n",
       "      0.4,\n",
       "      0.45,\n",
       "      0.5,\n",
       "      0.55,\n",
       "      0.6,\n",
       "      0.65,\n",
       "      0.7,\n",
       "      0.75,\n",
       "      0.8,\n",
       "      0.85,\n",
       "      0.9,\n",
       "      0.95,\n",
       "      1],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      3,\n",
       "      3,\n",
       "      8,\n",
       "      26,\n",
       "      16,\n",
       "      26,\n",
       "      17,\n",
       "      13,\n",
       "      5,\n",
       "      3,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeDepth': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [11,\n",
       "      67,\n",
       "      641,\n",
       "      3138,\n",
       "      6868,\n",
       "      12475,\n",
       "      21477,\n",
       "      17174,\n",
       "      5825,\n",
       "      1989,\n",
       "      945,\n",
       "      476,\n",
       "      212,\n",
       "      94,\n",
       "      59,\n",
       "      30,\n",
       "      19,\n",
       "      11,\n",
       "      2,\n",
       "      4],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 13},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      8,\n",
       "      13,\n",
       "      25,\n",
       "      29,\n",
       "      15,\n",
       "      10,\n",
       "      9,\n",
       "      8,\n",
       "      3,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeQuality': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [42,\n",
       "      34,\n",
       "      79,\n",
       "      100,\n",
       "      20247,\n",
       "      14345,\n",
       "      9795,\n",
       "      6614,\n",
       "      5441,\n",
       "      3845,\n",
       "      2497,\n",
       "      2219,\n",
       "      1635,\n",
       "      1101,\n",
       "      926,\n",
       "      674,\n",
       "      461,\n",
       "      380,\n",
       "      262,\n",
       "      833],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      1,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      119],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'siteQualityMetrics': [{'metric': 'FS', 'value': 1.1223795673810724},\n",
       "    {'metric': 'InbreedingCoeff', 'value': -0.0008395130824121466},\n",
       "    {'metric': 'MQ', 'value': 59.99216274475224},\n",
       "    {'metric': 'MQRankSum', 'value': 0},\n",
       "    {'metric': 'QD', 'value': 13.505564},\n",
       "    {'metric': 'ReadPosRankSum', 'value': 0.345},\n",
       "    {'metric': 'SiteQuality', 'value': 100738},\n",
       "    {'metric': 'SOR', 'value': 0.804},\n",
       "    {'metric': 'VQSLOD', 'value': None}]}},\n",
       " 'flags': [],\n",
       " 'rsid': 'rs121965019',\n",
       " 'sortedTranscriptConsequences': [{'canonical': True,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'p.Trp402Ter',\n",
       "   'hgvsc': 'c.1205G>A',\n",
       "   'hgvsp': 'p.Trp402Ter',\n",
       "   'lof': 'HC',\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'stop_gained',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000247933',\n",
       "   'transcript_version': '8'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'p.Trp270Ter',\n",
       "   'hgvsc': 'c.809G>A',\n",
       "   'hgvsp': 'p.Trp270Ter',\n",
       "   'lof': 'HC',\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'stop_gained',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000514224',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.7G>A',\n",
       "   'hgvsc': 'n.7G>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000502829',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.1312G>A',\n",
       "   'hgvsc': 'n.1312G>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000514698',\n",
       "   'transcript_version': '5'}]}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Allele Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>African</th>\n",
       "      <td>0.0</td>\n",
       "      <td>8642.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3654.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4988.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino</th>\n",
       "      <td>0.0</td>\n",
       "      <td>842.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>410.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>432.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <td>0.0</td>\n",
       "      <td>286.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1558.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Japanese</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Korean</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Other</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>538.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1020.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3446.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1838.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1608.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <td>2.0</td>\n",
       "      <td>15334.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Bulgarian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Estonian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4524.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) North-western</th>\n",
       "      <td>2.0</td>\n",
       "      <td>8572.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Other</th>\n",
       "      <td>0.0</td>\n",
       "      <td>2134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Southern</th>\n",
       "      <td>0.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Swedish</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Female</th>\n",
       "      <td>2.0</td>\n",
       "      <td>6762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>8572.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1086.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>562.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>524.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian Female</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total Female</th>\n",
       "      <td>2.0</td>\n",
       "      <td>13842.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total Male</th>\n",
       "      <td>0.0</td>\n",
       "      <td>17352.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>2.0</td>\n",
       "      <td>31194.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000064</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Allele Count  Allele Number  \\\n",
       "African                                        0.0         8642.0   \n",
       "African Female                                 0.0         3654.0   \n",
       "African Male                                   0.0         4988.0   \n",
       "Latino                                         0.0          842.0   \n",
       "Latino Female                                  0.0          410.0   \n",
       "Latino Male                                    0.0          432.0   \n",
       "Ashkenazi Jewish                               0.0          286.0   \n",
       "Ashkenazi Jewish Female                        0.0           78.0   \n",
       "Ashkenazi Jewish Male                          0.0          208.0   \n",
       "East Asian                                     0.0         1558.0   \n",
       "East Asian Japanese                            0.0            0.0   \n",
       "East Asian Korean                              0.0            0.0   \n",
       "East Asian Other                               0.0            0.0   \n",
       "East Asian Female                              0.0          538.0   \n",
       "East Asian Male                                0.0         1020.0   \n",
       "European (Finnish)                             0.0         3446.0   \n",
       "European (Finnish) Female                      0.0         1838.0   \n",
       "European (Finnish) Male                        0.0         1608.0   \n",
       "European (non-Finnish)                         2.0        15334.0   \n",
       "European (non-Finnish) Bulgarian               0.0            0.0   \n",
       "European (non-Finnish) Estonian                0.0         4524.0   \n",
       "European (non-Finnish) North-western           2.0         8572.0   \n",
       "European (non-Finnish) Other                   0.0         2134.0   \n",
       "European (non-Finnish) Southern                0.0          104.0   \n",
       "European (non-Finnish) Swedish                 0.0            0.0   \n",
       "European (non-Finnish) Female                  2.0         6762.0   \n",
       "European (non-Finnish) Male                    0.0         8572.0   \n",
       "Other                                          0.0         1086.0   \n",
       "Other Female                                   0.0          562.0   \n",
       "Other Male                                     0.0          524.0   \n",
       "South Asian                                    0.0            0.0   \n",
       "South Asian Female                             0.0            0.0   \n",
       "South Asian Male                               0.0            0.0   \n",
       "Total Female                                   2.0        13842.0   \n",
       "Total Male                                     0.0        17352.0   \n",
       "Total                                          2.0        31194.0   \n",
       "\n",
       "                                      Number of Homozygotes  Allele Frequency  \n",
       "African                                                 0.0          0.000000  \n",
       "African Female                                          0.0          0.000000  \n",
       "African Male                                            0.0          0.000000  \n",
       "Latino                                                  0.0          0.000000  \n",
       "Latino Female                                           0.0          0.000000  \n",
       "Latino Male                                             0.0          0.000000  \n",
       "Ashkenazi Jewish                                        0.0          0.000000  \n",
       "Ashkenazi Jewish Female                                 0.0          0.000000  \n",
       "Ashkenazi Jewish Male                                   0.0          0.000000  \n",
       "East Asian                                              0.0          0.000000  \n",
       "East Asian Japanese                                     0.0          0.000000  \n",
       "East Asian Korean                                       0.0          0.000000  \n",
       "East Asian Other                                        0.0          0.000000  \n",
       "East Asian Female                                       0.0          0.000000  \n",
       "East Asian Male                                         0.0          0.000000  \n",
       "European (Finnish)                                      0.0          0.000000  \n",
       "European (Finnish) Female                               0.0          0.000000  \n",
       "European (Finnish) Male                                 0.0          0.000000  \n",
       "European (non-Finnish)                                  0.0          0.000130  \n",
       "European (non-Finnish) Bulgarian                        0.0          0.000000  \n",
       "European (non-Finnish) Estonian                         0.0          0.000000  \n",
       "European (non-Finnish) North-western                    0.0          0.000233  \n",
       "European (non-Finnish) Other                            0.0          0.000000  \n",
       "European (non-Finnish) Southern                         0.0          0.000000  \n",
       "European (non-Finnish) Swedish                          0.0          0.000000  \n",
       "European (non-Finnish) Female                           0.0          0.000296  \n",
       "European (non-Finnish) Male                             0.0          0.000000  \n",
       "Other                                                   0.0          0.000000  \n",
       "Other Female                                            0.0          0.000000  \n",
       "Other Male                                              0.0          0.000000  \n",
       "South Asian                                             0.0          0.000000  \n",
       "South Asian Female                                      0.0          0.000000  \n",
       "South Asian Male                                        0.0          0.000000  \n",
       "Total Female                                            0.0          0.000144  \n",
       "Total Male                                              0.0          0.000000  \n",
       "Total                                                   0.0          0.000064  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vs = VariantSearch(2, '4-980801-G-A')\n",
    "df, meta = vs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
