{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pynoma import VariantSearch\n",
    "vs = VariantSearch(3, '4-1002747-G-A')\n",
    "js=vs.get_json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'variant': {'variantId': '4-1002747-G-A',\n",
       "   'reference_genome': 'GRCh38',\n",
       "   'chrom': '4',\n",
       "   'pos': 1002747,\n",
       "   'ref': 'G',\n",
       "   'alt': 'A',\n",
       "   'colocatedVariants': ['4-1002747-G-T'],\n",
       "   'multiNucleotideVariants': [],\n",
       "   'exome': None,\n",
       "   'genome': {'ac': 120,\n",
       "    'an': 142464,\n",
       "    'ac_hemi': 0,\n",
       "    'ac_hom': 0,\n",
       "    'faf95': {'popmax': 0.0013063199999999997, 'popmax_population': 'NFE'},\n",
       "    'filters': [],\n",
       "    'populations': [{'id': 'AFR',\n",
       "      'ac': 13,\n",
       "      'an': 41628,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0},\n",
       "     {'id': 'AFR_FEMALE', 'ac': 6, 'an': 22480, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AFR_MALE', 'ac': 7, 'an': 19148, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMI_FEMALE', 'ac': 0, 'an': 470, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMI_MALE', 'ac': 0, 'an': 428, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMR', 'ac': 3, 'an': 13600, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMR_FEMALE', 'ac': 2, 'an': 5882, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMR_MALE', 'ac': 1, 'an': 7718, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'ASJ', 'ac': 0, 'an': 3316, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'ASJ_FEMALE', 'ac': 0, 'an': 1756, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'ASJ_MALE', 'ac': 0, 'an': 1560, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'EAS', 'ac': 0, 'an': 3124, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'EAS_FEMALE', 'ac': 0, 'an': 1440, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'EAS_MALE', 'ac': 0, 'an': 1684, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'FIN', 'ac': 3, 'an': 10322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'FIN_FEMALE', 'ac': 1, 'an': 2476, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'FIN_MALE', 'ac': 2, 'an': 7846, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'NFE', 'ac': 100, 'an': 64398, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'NFE_FEMALE', 'ac': 59, 'an': 37292, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'NFE_MALE', 'ac': 41, 'an': 27106, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'OTH', 'ac': 1, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'OTH_FEMALE', 'ac': 1, 'an': 1094, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'OTH_MALE', 'ac': 0, 'an': 1048, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'SAS', 'ac': 0, 'an': 3036, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'SAS_FEMALE', 'ac': 0, 'an': 560, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'SAS_MALE', 'ac': 0, 'an': 2476, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'FEMALE', 'ac': 69, 'an': 73450, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'MALE', 'ac': 51, 'an': 69014, 'ac_hemi': 0, 'ac_hom': 0}],\n",
       "    'age_distribution': None,\n",
       "    'qualityMetrics': {'alleleBalance': {'alt': {'bin_edges': [0,\n",
       "        0.05,\n",
       "        0.1,\n",
       "        0.15,\n",
       "        0.2,\n",
       "        0.25,\n",
       "        0.3,\n",
       "        0.35,\n",
       "        0.4,\n",
       "        0.45,\n",
       "        0.5,\n",
       "        0.55,\n",
       "        0.6,\n",
       "        0.65,\n",
       "        0.7,\n",
       "        0.75,\n",
       "        0.8,\n",
       "        0.85,\n",
       "        0.9,\n",
       "        0.95,\n",
       "        1],\n",
       "       'bin_freq': [0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        3,\n",
       "        3,\n",
       "        8,\n",
       "        26,\n",
       "        16,\n",
       "        26,\n",
       "        17,\n",
       "        13,\n",
       "        5,\n",
       "        3,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0],\n",
       "       'n_smaller': 0,\n",
       "       'n_larger': 0}},\n",
       "     'genotypeDepth': {'all': {'bin_edges': [0,\n",
       "        5,\n",
       "        10,\n",
       "        15,\n",
       "        20,\n",
       "        25,\n",
       "        30,\n",
       "        35,\n",
       "        40,\n",
       "        45,\n",
       "        50,\n",
       "        55,\n",
       "        60,\n",
       "        65,\n",
       "        70,\n",
       "        75,\n",
       "        80,\n",
       "        85,\n",
       "        90,\n",
       "        95,\n",
       "        100],\n",
       "       'bin_freq': [11,\n",
       "        67,\n",
       "        641,\n",
       "        3138,\n",
       "        6868,\n",
       "        12475,\n",
       "        21477,\n",
       "        17174,\n",
       "        5825,\n",
       "        1989,\n",
       "        945,\n",
       "        476,\n",
       "        212,\n",
       "        94,\n",
       "        59,\n",
       "        30,\n",
       "        19,\n",
       "        11,\n",
       "        2,\n",
       "        4],\n",
       "       'n_smaller': 0,\n",
       "       'n_larger': 13},\n",
       "      'alt': {'bin_edges': [0,\n",
       "        5,\n",
       "        10,\n",
       "        15,\n",
       "        20,\n",
       "        25,\n",
       "        30,\n",
       "        35,\n",
       "        40,\n",
       "        45,\n",
       "        50,\n",
       "        55,\n",
       "        60,\n",
       "        65,\n",
       "        70,\n",
       "        75,\n",
       "        80,\n",
       "        85,\n",
       "        90,\n",
       "        95,\n",
       "        100],\n",
       "       'bin_freq': [0,\n",
       "        0,\n",
       "        0,\n",
       "        8,\n",
       "        13,\n",
       "        25,\n",
       "        29,\n",
       "        15,\n",
       "        10,\n",
       "        9,\n",
       "        8,\n",
       "        3,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0],\n",
       "       'n_smaller': 0,\n",
       "       'n_larger': 0}},\n",
       "     'genotypeQuality': {'all': {'bin_edges': [0,\n",
       "        5,\n",
       "        10,\n",
       "        15,\n",
       "        20,\n",
       "        25,\n",
       "        30,\n",
       "        35,\n",
       "        40,\n",
       "        45,\n",
       "        50,\n",
       "        55,\n",
       "        60,\n",
       "        65,\n",
       "        70,\n",
       "        75,\n",
       "        80,\n",
       "        85,\n",
       "        90,\n",
       "        95,\n",
       "        100],\n",
       "       'bin_freq': [42,\n",
       "        34,\n",
       "        79,\n",
       "        100,\n",
       "        20247,\n",
       "        14345,\n",
       "        9795,\n",
       "        6614,\n",
       "        5441,\n",
       "        3845,\n",
       "        2497,\n",
       "        2219,\n",
       "        1635,\n",
       "        1101,\n",
       "        926,\n",
       "        674,\n",
       "        461,\n",
       "        380,\n",
       "        262,\n",
       "        833],\n",
       "       'n_smaller': 0,\n",
       "       'n_larger': 0},\n",
       "      'alt': {'bin_edges': [0,\n",
       "        5,\n",
       "        10,\n",
       "        15,\n",
       "        20,\n",
       "        25,\n",
       "        30,\n",
       "        35,\n",
       "        40,\n",
       "        45,\n",
       "        50,\n",
       "        55,\n",
       "        60,\n",
       "        65,\n",
       "        70,\n",
       "        75,\n",
       "        80,\n",
       "        85,\n",
       "        90,\n",
       "        95,\n",
       "        100],\n",
       "       'bin_freq': [0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        1,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        119],\n",
       "       'n_smaller': 0,\n",
       "       'n_larger': 0}},\n",
       "     'siteQualityMetrics': [{'metric': 'FS', 'value': 1.1223795673810724},\n",
       "      {'metric': 'InbreedingCoeff', 'value': -0.0008395130824121466},\n",
       "      {'metric': 'MQ', 'value': 59.99216274475224},\n",
       "      {'metric': 'MQRankSum', 'value': 0},\n",
       "      {'metric': 'QD', 'value': 13.505564},\n",
       "      {'metric': 'ReadPosRankSum', 'value': 0.345},\n",
       "      {'metric': 'SiteQuality', 'value': 100738},\n",
       "      {'metric': 'SOR', 'value': 0.804},\n",
       "      {'metric': 'VQSLOD', 'value': None}]}},\n",
       "   'flags': [],\n",
       "   'rsid': 'rs121965019',\n",
       "   'sortedTranscriptConsequences': [{'canonical': True,\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_version': '12',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Trp402Ter',\n",
       "     'hgvsc': 'c.1205G>A',\n",
       "     'hgvsp': 'p.Trp402Ter',\n",
       "     'lof': 'HC',\n",
       "     'lof_flags': None,\n",
       "     'lof_filter': None,\n",
       "     'major_consequence': 'stop_gained',\n",
       "     'polyphen_prediction': None,\n",
       "     'sift_prediction': None,\n",
       "     'transcript_id': 'ENST00000247933',\n",
       "     'transcript_version': '8'},\n",
       "    {'canonical': None,\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_version': '12',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Trp270Ter',\n",
       "     'hgvsc': 'c.809G>A',\n",
       "     'hgvsp': 'p.Trp270Ter',\n",
       "     'lof': 'HC',\n",
       "     'lof_flags': None,\n",
       "     'lof_filter': None,\n",
       "     'major_consequence': 'stop_gained',\n",
       "     'polyphen_prediction': None,\n",
       "     'sift_prediction': None,\n",
       "     'transcript_id': 'ENST00000514224',\n",
       "     'transcript_version': '1'},\n",
       "    {'canonical': None,\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_version': '12',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'n.7G>A',\n",
       "     'hgvsc': 'n.7G>A',\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_flags': None,\n",
       "     'lof_filter': None,\n",
       "     'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "     'polyphen_prediction': None,\n",
       "     'sift_prediction': None,\n",
       "     'transcript_id': 'ENST00000502829',\n",
       "     'transcript_version': '1'},\n",
       "    {'canonical': None,\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_version': '12',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'n.1312G>A',\n",
       "     'hgvsc': 'n.1312G>A',\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_flags': None,\n",
       "     'lof_filter': None,\n",
       "     'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "     'polyphen_prediction': None,\n",
       "     'sift_prediction': None,\n",
       "     'transcript_id': 'ENST00000514698',\n",
       "     'transcript_version': '5'}]}}}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "js"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>data.variant.variantId</th>\n",
       "      <th>data.variant.reference_genome</th>\n",
       "      <th>data.variant.chrom</th>\n",
       "      <th>data.variant.pos</th>\n",
       "      <th>data.variant.ref</th>\n",
       "      <th>data.variant.alt</th>\n",
       "      <th>data.variant.colocatedVariants</th>\n",
       "      <th>data.variant.multiNucleotideVariants</th>\n",
       "      <th>data.variant.exome</th>\n",
       "      <th>data.variant.genome.ac</th>\n",
       "      <th>...</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.pab_max</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.QD</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.ReadPosRankSum</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.RF</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.SiteQuality</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.SOR</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.VQSLOD</th>\n",
       "      <th>data.variant.flags</th>\n",
       "      <th>data.variant.rsid</th>\n",
       "      <th>data.variant.sortedTranscriptConsequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>4</td>\n",
       "      <td>1002747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>[4-1002747-G-T]</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>120</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>13.505564</td>\n",
       "      <td>0.345</td>\n",
       "      <td>None</td>\n",
       "      <td>100738</td>\n",
       "      <td>0.804</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>rs121965019</td>\n",
       "      <td>[{'canonical': True, 'gene_id': 'ENSG000001274...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  data.variant.variantId data.variant.reference_genome data.variant.chrom  \\\n",
       "0          4-1002747-G-A                        GRCh38                  4   \n",
       "\n",
       "   data.variant.pos data.variant.ref data.variant.alt  \\\n",
       "0           1002747                G                A   \n",
       "\n",
       "  data.variant.colocatedVariants data.variant.multiNucleotideVariants  \\\n",
       "0                [4-1002747-G-T]                                   []   \n",
       "\n",
       "  data.variant.exome  data.variant.genome.ac  ...  \\\n",
       "0               None                     120  ...   \n",
       "\n",
       "   data.variant.genome.qualityMetrics.siteQualityMetrics.pab_max  \\\n",
       "0                                               None               \n",
       "\n",
       "   data.variant.genome.qualityMetrics.siteQualityMetrics.QD  \\\n",
       "0                                          13.505564          \n",
       "\n",
       "   data.variant.genome.qualityMetrics.siteQualityMetrics.ReadPosRankSum  \\\n",
       "0                                              0.345                      \n",
       "\n",
       "   data.variant.genome.qualityMetrics.siteQualityMetrics.RF  \\\n",
       "0                                               None          \n",
       "\n",
       "  data.variant.genome.qualityMetrics.siteQualityMetrics.SiteQuality  \\\n",
       "0                                             100738                  \n",
       "\n",
       "  data.variant.genome.qualityMetrics.siteQualityMetrics.SOR  \\\n",
       "0                                              0.804          \n",
       "\n",
       "  data.variant.genome.qualityMetrics.siteQualityMetrics.VQSLOD  \\\n",
       "0                                               None             \n",
       "\n",
       "  data.variant.flags data.variant.rsid  \\\n",
       "0                 []       rs121965019   \n",
       "\n",
       "           data.variant.sortedTranscriptConsequences  \n",
       "0  [{'canonical': True, 'gene_id': 'ENSG000001274...  \n",
       "\n",
       "[1 rows x 55 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.json_normalize(js)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant.variantId</th>\n",
       "      <th>variant.reference_genome</th>\n",
       "      <th>variant.chrom</th>\n",
       "      <th>variant.pos</th>\n",
       "      <th>variant.ref</th>\n",
       "      <th>variant.alt</th>\n",
       "      <th>variant.colocatedVariants</th>\n",
       "      <th>variant.multiNucleotideVariants</th>\n",
       "      <th>variant.exome</th>\n",
       "      <th>variant.genome.ac</th>\n",
       "      <th>...</th>\n",
       "      <th>variant.genome.qualityMetrics.genotypeQuality.all.n_smaller</th>\n",
       "      <th>variant.genome.qualityMetrics.genotypeQuality.all.n_larger</th>\n",
       "      <th>variant.genome.qualityMetrics.genotypeQuality.alt.bin_edges</th>\n",
       "      <th>variant.genome.qualityMetrics.genotypeQuality.alt.bin_freq</th>\n",
       "      <th>variant.genome.qualityMetrics.genotypeQuality.alt.n_smaller</th>\n",
       "      <th>variant.genome.qualityMetrics.genotypeQuality.alt.n_larger</th>\n",
       "      <th>variant.genome.qualityMetrics.siteQualityMetrics</th>\n",
       "      <th>variant.flags</th>\n",
       "      <th>variant.rsid</th>\n",
       "      <th>variant.sortedTranscriptConsequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>4</td>\n",
       "      <td>1002747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>[4-1002747-G-T]</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>120</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55,...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, ...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[{'metric': 'FS', 'value': 1.1223795673810724}...</td>\n",
       "      <td>[]</td>\n",
       "      <td>rs121965019</td>\n",
       "      <td>[{'canonical': True, 'gene_id': 'ENSG000001274...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  variant.variantId variant.reference_genome variant.chrom  variant.pos  \\\n",
       "0     4-1002747-G-A                   GRCh38             4      1002747   \n",
       "\n",
       "  variant.ref variant.alt variant.colocatedVariants  \\\n",
       "0           G           A           [4-1002747-G-T]   \n",
       "\n",
       "  variant.multiNucleotideVariants variant.exome  variant.genome.ac  ...  \\\n",
       "0                              []          None                120  ...   \n",
       "\n",
       "   variant.genome.qualityMetrics.genotypeQuality.all.n_smaller  \\\n",
       "0                                                  0             \n",
       "\n",
       "   variant.genome.qualityMetrics.genotypeQuality.all.n_larger  \\\n",
       "0                                                  0            \n",
       "\n",
       "   variant.genome.qualityMetrics.genotypeQuality.alt.bin_edges  \\\n",
       "0  [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55,...             \n",
       "\n",
       "   variant.genome.qualityMetrics.genotypeQuality.alt.bin_freq  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, ...            \n",
       "\n",
       "  variant.genome.qualityMetrics.genotypeQuality.alt.n_smaller  \\\n",
       "0                                                  0            \n",
       "\n",
       "  variant.genome.qualityMetrics.genotypeQuality.alt.n_larger  \\\n",
       "0                                                  0           \n",
       "\n",
       "    variant.genome.qualityMetrics.siteQualityMetrics variant.flags  \\\n",
       "0  [{'metric': 'FS', 'value': 1.1223795673810724}...            []   \n",
       "\n",
       "  variant.rsid               variant.sortedTranscriptConsequences  \n",
       "0  rs121965019  [{'canonical': True, 'gene_id': 'ENSG000001274...  \n",
       "\n",
       "[1 rows x 42 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.json_normalize(js['data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ac': 120,\n",
       " 'an': 142464,\n",
       " 'ac_hemi': 0,\n",
       " 'ac_hom': 0,\n",
       " 'faf95': {'popmax': 0.0013063199999999997, 'popmax_population': 'NFE'},\n",
       " 'filters': [],\n",
       " 'populations': [{'id': 'AFR',\n",
       "   'ac': 13,\n",
       "   'an': 41628,\n",
       "   'ac_hemi': 0,\n",
       "   'ac_hom': 0},\n",
       "  {'id': 'AFR_FEMALE', 'ac': 6, 'an': 22480, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'AFR_MALE', 'ac': 7, 'an': 19148, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'AMI_FEMALE', 'ac': 0, 'an': 470, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'AMI_MALE', 'ac': 0, 'an': 428, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'AMR', 'ac': 3, 'an': 13600, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'AMR_FEMALE', 'ac': 2, 'an': 5882, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'AMR_MALE', 'ac': 1, 'an': 7718, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'ASJ', 'ac': 0, 'an': 3316, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'ASJ_FEMALE', 'ac': 0, 'an': 1756, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'ASJ_MALE', 'ac': 0, 'an': 1560, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'EAS', 'ac': 0, 'an': 3124, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'EAS_FEMALE', 'ac': 0, 'an': 1440, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'EAS_MALE', 'ac': 0, 'an': 1684, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'FIN', 'ac': 3, 'an': 10322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'FIN_FEMALE', 'ac': 1, 'an': 2476, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'FIN_MALE', 'ac': 2, 'an': 7846, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'NFE', 'ac': 100, 'an': 64398, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'NFE_FEMALE', 'ac': 59, 'an': 37292, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'NFE_MALE', 'ac': 41, 'an': 27106, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'OTH', 'ac': 1, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'OTH_FEMALE', 'ac': 1, 'an': 1094, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'OTH_MALE', 'ac': 0, 'an': 1048, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'SAS', 'ac': 0, 'an': 3036, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'SAS_FEMALE', 'ac': 0, 'an': 560, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'SAS_MALE', 'ac': 0, 'an': 2476, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'FEMALE', 'ac': 69, 'an': 73450, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'MALE', 'ac': 51, 'an': 69014, 'ac_hemi': 0, 'ac_hom': 0}],\n",
       " 'age_distribution': None,\n",
       " 'qualityMetrics': {'alleleBalance': {'alt': {'bin_edges': [0,\n",
       "     0.05,\n",
       "     0.1,\n",
       "     0.15,\n",
       "     0.2,\n",
       "     0.25,\n",
       "     0.3,\n",
       "     0.35,\n",
       "     0.4,\n",
       "     0.45,\n",
       "     0.5,\n",
       "     0.55,\n",
       "     0.6,\n",
       "     0.65,\n",
       "     0.7,\n",
       "     0.75,\n",
       "     0.8,\n",
       "     0.85,\n",
       "     0.9,\n",
       "     0.95,\n",
       "     1],\n",
       "    'bin_freq': [0,\n",
       "     0,\n",
       "     0,\n",
       "     0,\n",
       "     0,\n",
       "     3,\n",
       "     3,\n",
       "     8,\n",
       "     26,\n",
       "     16,\n",
       "     26,\n",
       "     17,\n",
       "     13,\n",
       "     5,\n",
       "     3,\n",
       "     0,\n",
       "     0,\n",
       "     0,\n",
       "     0,\n",
       "     0],\n",
       "    'n_smaller': 0,\n",
       "    'n_larger': 0}},\n",
       "  'genotypeDepth': {'all': {'bin_edges': [0,\n",
       "     5,\n",
       "     10,\n",
       "     15,\n",
       "     20,\n",
       "     25,\n",
       "     30,\n",
       "     35,\n",
       "     40,\n",
       "     45,\n",
       "     50,\n",
       "     55,\n",
       "     60,\n",
       "     65,\n",
       "     70,\n",
       "     75,\n",
       "     80,\n",
       "     85,\n",
       "     90,\n",
       "     95,\n",
       "     100],\n",
       "    'bin_freq': [11,\n",
       "     67,\n",
       "     641,\n",
       "     3138,\n",
       "     6868,\n",
       "     12475,\n",
       "     21477,\n",
       "     17174,\n",
       "     5825,\n",
       "     1989,\n",
       "     945,\n",
       "     476,\n",
       "     212,\n",
       "     94,\n",
       "     59,\n",
       "     30,\n",
       "     19,\n",
       "     11,\n",
       "     2,\n",
       "     4],\n",
       "    'n_smaller': 0,\n",
       "    'n_larger': 13},\n",
       "   'alt': {'bin_edges': [0,\n",
       "     5,\n",
       "     10,\n",
       "     15,\n",
       "     20,\n",
       "     25,\n",
       "     30,\n",
       "     35,\n",
       "     40,\n",
       "     45,\n",
       "     50,\n",
       "     55,\n",
       "     60,\n",
       "     65,\n",
       "     70,\n",
       "     75,\n",
       "     80,\n",
       "     85,\n",
       "     90,\n",
       "     95,\n",
       "     100],\n",
       "    'bin_freq': [0,\n",
       "     0,\n",
       "     0,\n",
       "     8,\n",
       "     13,\n",
       "     25,\n",
       "     29,\n",
       "     15,\n",
       "     10,\n",
       "     9,\n",
       "     8,\n",
       "     3,\n",
       "     0,\n",
       "     0,\n",
       "     0,\n",
       "     0,\n",
       "     0,\n",
       "     0,\n",
       "     0,\n",
       "     0],\n",
       "    'n_smaller': 0,\n",
       "    'n_larger': 0}},\n",
       "  'genotypeQuality': {'all': {'bin_edges': [0,\n",
       "     5,\n",
       "     10,\n",
       "     15,\n",
       "     20,\n",
       "     25,\n",
       "     30,\n",
       "     35,\n",
       "     40,\n",
       "     45,\n",
       "     50,\n",
       "     55,\n",
       "     60,\n",
       "     65,\n",
       "     70,\n",
       "     75,\n",
       "     80,\n",
       "     85,\n",
       "     90,\n",
       "     95,\n",
       "     100],\n",
       "    'bin_freq': [42,\n",
       "     34,\n",
       "     79,\n",
       "     100,\n",
       "     20247,\n",
       "     14345,\n",
       "     9795,\n",
       "     6614,\n",
       "     5441,\n",
       "     3845,\n",
       "     2497,\n",
       "     2219,\n",
       "     1635,\n",
       "     1101,\n",
       "     926,\n",
       "     674,\n",
       "     461,\n",
       "     380,\n",
       "     262,\n",
       "     833],\n",
       "    'n_smaller': 0,\n",
       "    'n_larger': 0},\n",
       "   'alt': {'bin_edges': [0,\n",
       "     5,\n",
       "     10,\n",
       "     15,\n",
       "     20,\n",
       "     25,\n",
       "     30,\n",
       "     35,\n",
       "     40,\n",
       "     45,\n",
       "     50,\n",
       "     55,\n",
       "     60,\n",
       "     65,\n",
       "     70,\n",
       "     75,\n",
       "     80,\n",
       "     85,\n",
       "     90,\n",
       "     95,\n",
       "     100],\n",
       "    'bin_freq': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 119],\n",
       "    'n_smaller': 0,\n",
       "    'n_larger': 0}},\n",
       "  'siteQualityMetrics': [{'metric': 'FS', 'value': 1.1223795673810724},\n",
       "   {'metric': 'InbreedingCoeff', 'value': -0.0008395130824121466},\n",
       "   {'metric': 'MQ', 'value': 59.99216274475224},\n",
       "   {'metric': 'MQRankSum', 'value': 0},\n",
       "   {'metric': 'QD', 'value': 13.505564},\n",
       "   {'metric': 'ReadPosRankSum', 'value': 0.345},\n",
       "   {'metric': 'SiteQuality', 'value': 100738},\n",
       "   {'metric': 'SOR', 'value': 0.804},\n",
       "   {'metric': 'VQSLOD', 'value': None}]}}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "js['data']['variant']['genome']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ac</th>\n",
       "      <th>an</th>\n",
       "      <th>ac_hemi</th>\n",
       "      <th>ac_hom</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AFR</th>\n",
       "      <td>13</td>\n",
       "      <td>41628</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFR_FEMALE</th>\n",
       "      <td>6</td>\n",
       "      <td>22480</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFR_MALE</th>\n",
       "      <td>7</td>\n",
       "      <td>19148</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMI</th>\n",
       "      <td>0</td>\n",
       "      <td>898</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMI_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>470</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMI_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>428</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMR</th>\n",
       "      <td>3</td>\n",
       "      <td>13600</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMR_FEMALE</th>\n",
       "      <td>2</td>\n",
       "      <td>5882</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMR_MALE</th>\n",
       "      <td>1</td>\n",
       "      <td>7718</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASJ</th>\n",
       "      <td>0</td>\n",
       "      <td>3316</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASJ_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>1756</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASJ_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>1560</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAS</th>\n",
       "      <td>0</td>\n",
       "      <td>3124</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAS_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>1440</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAS_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>1684</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIN</th>\n",
       "      <td>3</td>\n",
       "      <td>10322</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIN_FEMALE</th>\n",
       "      <td>1</td>\n",
       "      <td>2476</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIN_MALE</th>\n",
       "      <td>2</td>\n",
       "      <td>7846</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE</th>\n",
       "      <td>100</td>\n",
       "      <td>64398</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE_FEMALE</th>\n",
       "      <td>59</td>\n",
       "      <td>37292</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFE_MALE</th>\n",
       "      <td>41</td>\n",
       "      <td>27106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTH</th>\n",
       "      <td>1</td>\n",
       "      <td>2142</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTH_FEMALE</th>\n",
       "      <td>1</td>\n",
       "      <td>1094</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTH_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>1048</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAS</th>\n",
       "      <td>0</td>\n",
       "      <td>3036</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAS_FEMALE</th>\n",
       "      <td>0</td>\n",
       "      <td>560</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAS_MALE</th>\n",
       "      <td>0</td>\n",
       "      <td>2476</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FEMALE</th>\n",
       "      <td>69</td>\n",
       "      <td>73450</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MALE</th>\n",
       "      <td>51</td>\n",
       "      <td>69014</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             ac     an  ac_hemi  ac_hom\n",
       "id                                     \n",
       "AFR          13  41628        0       0\n",
       "AFR_FEMALE    6  22480        0       0\n",
       "AFR_MALE      7  19148        0       0\n",
       "AMI           0    898        0       0\n",
       "AMI_FEMALE    0    470        0       0\n",
       "AMI_MALE      0    428        0       0\n",
       "AMR           3  13600        0       0\n",
       "AMR_FEMALE    2   5882        0       0\n",
       "AMR_MALE      1   7718        0       0\n",
       "ASJ           0   3316        0       0\n",
       "ASJ_FEMALE    0   1756        0       0\n",
       "ASJ_MALE      0   1560        0       0\n",
       "EAS           0   3124        0       0\n",
       "EAS_FEMALE    0   1440        0       0\n",
       "EAS_MALE      0   1684        0       0\n",
       "FIN           3  10322        0       0\n",
       "FIN_FEMALE    1   2476        0       0\n",
       "FIN_MALE      2   7846        0       0\n",
       "NFE         100  64398        0       0\n",
       "NFE_FEMALE   59  37292        0       0\n",
       "NFE_MALE     41  27106        0       0\n",
       "OTH           1   2142        0       0\n",
       "OTH_FEMALE    1   1094        0       0\n",
       "OTH_MALE      0   1048        0       0\n",
       "SAS           0   3036        0       0\n",
       "SAS_FEMALE    0    560        0       0\n",
       "SAS_MALE      0   2476        0       0\n",
       "FEMALE       69  73450        0       0\n",
       "MALE         51  69014        0       0"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reqdf = pd.json_normalize(js['data']['variant']['genome']['populations']).set_index('id')\n",
    "reqdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "POPULATION_ID_MAP = {\n",
    "            'AFR': 'African',\n",
    "            'AMI': 'Amish',\n",
    "            'AMR': 'Latino',\n",
    "            'ASJ': 'Ashkenazi Jewish',\n",
    "            'EAS': 'East Asian',\n",
    "            'FIN': 'European (Finnish)',\n",
    "            'NFE': 'European (non-Finnish)',\n",
    "            'OTH': 'Other',\n",
    "            'SAS': 'South Asian'\n",
    "        }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "primary_table = {\n",
    "    \"Female Allele Count\",\n",
    "    \"Female Allele Number\",\n",
    "    \"Female Number of Homozygotes\",\n",
    "    \"Female Number of Hemizygotes\"\n",
    "    \"Male Allele Count\",\n",
    "    \"Male Allele Number\",\n",
    "    \"Male Number of Homozygotes\",\n",
    "    \"Male Number of Hemizygotes\",\n",
    "    \"Total Allele Count\",\n",
    "    \"Total Allele Number\",\n",
    "    \"Total Number of Homozygotes\",\n",
    "    \"Total Number of Hemizygotes\"\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "POPULATION_ID_MAP['FEMALE'] = 'Female'\n",
    "POPULATION_ID_MAP['MALE'] = 'Male'\n",
    "\n",
    "new_index = {}\n",
    "frequencies = []\n",
    "for row in reqdf.iterrows():\n",
    "    row_id = row[0]\n",
    "    new_row_name = \"\"\n",
    "    for piece in row_id.split('_'):\n",
    "        new_row_name += POPULATION_ID_MAP[piece] + \" \"\n",
    "    new_row_name = new_row_name[0:-1]\n",
    "    new_index[row_id] = new_row_name\n",
    "    frequencies.append(row[1]['ac']/row[1]['an'])\n",
    "\n",
    "new_index['FEMALE'] = 'Total Female'\n",
    "new_index['MALE'] = 'Total Male'\n",
    "new_columns = {'ac': 'Allele Count', 'an': 'Allele Number',\n",
    "               'ac_hemi': 'Number of Hemizygotes', 'ac_hom': 'Number of Homozygotes'}\n",
    "\n",
    "df = reqdf.rename(columns=new_columns, index=new_index)\n",
    "\n",
    "total_row = df.loc['Total Female'] + df.loc['Total Male']\n",
    "total_row.name = 'Total'\n",
    "df = df.append([total_row])\n",
    "\n",
    "frequencies.append(df.loc['Total']['Allele Count'] / df.loc['Total']['Allele Number'])\n",
    "df['Allele Frequency'] = frequencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Number of Hemizygotes</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Allele Frequency</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>African</th>\n",
       "      <td>13</td>\n",
       "      <td>41628</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African Female</th>\n",
       "      <td>6</td>\n",
       "      <td>22480</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000267</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African Male</th>\n",
       "      <td>7</td>\n",
       "      <td>19148</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amish</th>\n",
       "      <td>0</td>\n",
       "      <td>898</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amish Female</th>\n",
       "      <td>0</td>\n",
       "      <td>470</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amish Male</th>\n",
       "      <td>0</td>\n",
       "      <td>428</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino</th>\n",
       "      <td>3</td>\n",
       "      <td>13600</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino Female</th>\n",
       "      <td>2</td>\n",
       "      <td>5882</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino Male</th>\n",
       "      <td>1</td>\n",
       "      <td>7718</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <td>0</td>\n",
       "      <td>3316</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish Female</th>\n",
       "      <td>0</td>\n",
       "      <td>1756</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish Male</th>\n",
       "      <td>0</td>\n",
       "      <td>1560</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian</th>\n",
       "      <td>0</td>\n",
       "      <td>3124</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Female</th>\n",
       "      <td>0</td>\n",
       "      <td>1440</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Male</th>\n",
       "      <td>0</td>\n",
       "      <td>1684</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish)</th>\n",
       "      <td>3</td>\n",
       "      <td>10322</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) Female</th>\n",
       "      <td>1</td>\n",
       "      <td>2476</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) Male</th>\n",
       "      <td>2</td>\n",
       "      <td>7846</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <td>100</td>\n",
       "      <td>64398</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Female</th>\n",
       "      <td>59</td>\n",
       "      <td>37292</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Male</th>\n",
       "      <td>41</td>\n",
       "      <td>27106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>1</td>\n",
       "      <td>2142</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other Female</th>\n",
       "      <td>1</td>\n",
       "      <td>1094</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other Male</th>\n",
       "      <td>0</td>\n",
       "      <td>1048</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian</th>\n",
       "      <td>0</td>\n",
       "      <td>3036</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian Female</th>\n",
       "      <td>0</td>\n",
       "      <td>560</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian Male</th>\n",
       "      <td>0</td>\n",
       "      <td>2476</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total Female</th>\n",
       "      <td>69</td>\n",
       "      <td>73450</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total Male</th>\n",
       "      <td>51</td>\n",
       "      <td>69014</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000739</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>120</td>\n",
       "      <td>142464</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000842</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Allele Count  Allele Number  \\\n",
       "                                                             \n",
       "African                                  13          41628   \n",
       "African Female                            6          22480   \n",
       "African Male                              7          19148   \n",
       "Amish                                     0            898   \n",
       "Amish Female                              0            470   \n",
       "Amish Male                                0            428   \n",
       "Latino                                    3          13600   \n",
       "Latino Female                             2           5882   \n",
       "Latino Male                               1           7718   \n",
       "Ashkenazi Jewish                          0           3316   \n",
       "Ashkenazi Jewish Female                   0           1756   \n",
       "Ashkenazi Jewish Male                     0           1560   \n",
       "East Asian                                0           3124   \n",
       "East Asian Female                         0           1440   \n",
       "East Asian Male                           0           1684   \n",
       "European (Finnish)                        3          10322   \n",
       "European (Finnish) Female                 1           2476   \n",
       "European (Finnish) Male                   2           7846   \n",
       "European (non-Finnish)                  100          64398   \n",
       "European (non-Finnish) Female            59          37292   \n",
       "European (non-Finnish) Male              41          27106   \n",
       "Other                                     1           2142   \n",
       "Other Female                              1           1094   \n",
       "Other Male                                0           1048   \n",
       "South Asian                               0           3036   \n",
       "South Asian Female                        0            560   \n",
       "South Asian Male                          0           2476   \n",
       "Total Female                             69          73450   \n",
       "Total Male                               51          69014   \n",
       "Total                                   120         142464   \n",
       "\n",
       "                               Number of Hemizygotes  Number of Homozygotes  \\\n",
       "                                                                              \n",
       "African                                            0                      0   \n",
       "African Female                                     0                      0   \n",
       "African Male                                       0                      0   \n",
       "Amish                                              0                      0   \n",
       "Amish Female                                       0                      0   \n",
       "Amish Male                                         0                      0   \n",
       "Latino                                             0                      0   \n",
       "Latino Female                                      0                      0   \n",
       "Latino Male                                        0                      0   \n",
       "Ashkenazi Jewish                                   0                      0   \n",
       "Ashkenazi Jewish Female                            0                      0   \n",
       "Ashkenazi Jewish Male                              0                      0   \n",
       "East Asian                                         0                      0   \n",
       "East Asian Female                                  0                      0   \n",
       "East Asian Male                                    0                      0   \n",
       "European (Finnish)                                 0                      0   \n",
       "European (Finnish) Female                          0                      0   \n",
       "European (Finnish) Male                            0                      0   \n",
       "European (non-Finnish)                             0                      0   \n",
       "European (non-Finnish) Female                      0                      0   \n",
       "European (non-Finnish) Male                        0                      0   \n",
       "Other                                              0                      0   \n",
       "Other Female                                       0                      0   \n",
       "Other Male                                         0                      0   \n",
       "South Asian                                        0                      0   \n",
       "South Asian Female                                 0                      0   \n",
       "South Asian Male                                   0                      0   \n",
       "Total Female                                       0                      0   \n",
       "Total Male                                         0                      0   \n",
       "Total                                              0                      0   \n",
       "\n",
       "                               Allele Frequency  \n",
       "                                                 \n",
       "African                                0.000312  \n",
       "African Female                         0.000267  \n",
       "African Male                           0.000366  \n",
       "Amish                                  0.000000  \n",
       "Amish Female                           0.000000  \n",
       "Amish Male                             0.000000  \n",
       "Latino                                 0.000221  \n",
       "Latino Female                          0.000340  \n",
       "Latino Male                            0.000130  \n",
       "Ashkenazi Jewish                       0.000000  \n",
       "Ashkenazi Jewish Female                0.000000  \n",
       "Ashkenazi Jewish Male                  0.000000  \n",
       "East Asian                             0.000000  \n",
       "East Asian Female                      0.000000  \n",
       "East Asian Male                        0.000000  \n",
       "European (Finnish)                     0.000291  \n",
       "European (Finnish) Female              0.000404  \n",
       "European (Finnish) Male                0.000255  \n",
       "European (non-Finnish)                 0.001553  \n",
       "European (non-Finnish) Female          0.001582  \n",
       "European (non-Finnish) Male            0.001513  \n",
       "Other                                  0.000467  \n",
       "Other Female                           0.000914  \n",
       "Other Male                             0.000000  \n",
       "South Asian                            0.000000  \n",
       "South Asian Female                     0.000000  \n",
       "South Asian Male                       0.000000  \n",
       "Total Female                           0.000939  \n",
       "Total Male                             0.000739  \n",
       "Total                                  0.000842  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.index.name = ''\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['data'])"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "js.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['variant'])"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "js['data'].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['variantId', 'reference_genome', 'chrom', 'pos', 'ref', 'alt', 'colocatedVariants', 'multiNucleotideVariants', 'exome', 'genome', 'flags', 'rsid', 'sortedTranscriptConsequences'])"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "js['data']['variant'].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'variantId': '4-1002747-G-A',\n",
       " 'reference_genome': 'GRCh38',\n",
       " 'chrom': '4',\n",
       " 'pos': 1002747,\n",
       " 'ref': 'G',\n",
       " 'alt': 'A',\n",
       " 'colocatedVariants': ['4-1002747-G-T'],\n",
       " 'multiNucleotideVariants': [],\n",
       " 'exome': None,\n",
       " 'genome': {'ac': 120,\n",
       "  'an': 142464,\n",
       "  'ac_hemi': 0,\n",
       "  'ac_hom': 0,\n",
       "  'faf95': {'popmax': 0.0013063199999999997, 'popmax_population': 'NFE'},\n",
       "  'filters': [],\n",
       "  'populations': [{'id': 'AFR',\n",
       "    'ac': 13,\n",
       "    'an': 41628,\n",
       "    'ac_hemi': 0,\n",
       "    'ac_hom': 0},\n",
       "   {'id': 'AFR_FEMALE', 'ac': 6, 'an': 22480, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AFR_MALE', 'ac': 7, 'an': 19148, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AMI_FEMALE', 'ac': 0, 'an': 470, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AMI_MALE', 'ac': 0, 'an': 428, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AMR', 'ac': 3, 'an': 13600, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AMR_FEMALE', 'ac': 2, 'an': 5882, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AMR_MALE', 'ac': 1, 'an': 7718, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'ASJ', 'ac': 0, 'an': 3316, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'ASJ_FEMALE', 'ac': 0, 'an': 1756, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'ASJ_MALE', 'ac': 0, 'an': 1560, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'EAS', 'ac': 0, 'an': 3124, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'EAS_FEMALE', 'ac': 0, 'an': 1440, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'EAS_MALE', 'ac': 0, 'an': 1684, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'FIN', 'ac': 3, 'an': 10322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'FIN_FEMALE', 'ac': 1, 'an': 2476, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'FIN_MALE', 'ac': 2, 'an': 7846, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'NFE', 'ac': 100, 'an': 64398, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'NFE_FEMALE', 'ac': 59, 'an': 37292, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'NFE_MALE', 'ac': 41, 'an': 27106, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'OTH', 'ac': 1, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'OTH_FEMALE', 'ac': 1, 'an': 1094, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'OTH_MALE', 'ac': 0, 'an': 1048, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'SAS', 'ac': 0, 'an': 3036, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'SAS_FEMALE', 'ac': 0, 'an': 560, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'SAS_MALE', 'ac': 0, 'an': 2476, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'FEMALE', 'ac': 69, 'an': 73450, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'MALE', 'ac': 51, 'an': 69014, 'ac_hemi': 0, 'ac_hom': 0}],\n",
       "  'age_distribution': None,\n",
       "  'qualityMetrics': {'alleleBalance': {'alt': {'bin_edges': [0,\n",
       "      0.05,\n",
       "      0.1,\n",
       "      0.15,\n",
       "      0.2,\n",
       "      0.25,\n",
       "      0.3,\n",
       "      0.35,\n",
       "      0.4,\n",
       "      0.45,\n",
       "      0.5,\n",
       "      0.55,\n",
       "      0.6,\n",
       "      0.65,\n",
       "      0.7,\n",
       "      0.75,\n",
       "      0.8,\n",
       "      0.85,\n",
       "      0.9,\n",
       "      0.95,\n",
       "      1],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      3,\n",
       "      3,\n",
       "      8,\n",
       "      26,\n",
       "      16,\n",
       "      26,\n",
       "      17,\n",
       "      13,\n",
       "      5,\n",
       "      3,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeDepth': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [11,\n",
       "      67,\n",
       "      641,\n",
       "      3138,\n",
       "      6868,\n",
       "      12475,\n",
       "      21477,\n",
       "      17174,\n",
       "      5825,\n",
       "      1989,\n",
       "      945,\n",
       "      476,\n",
       "      212,\n",
       "      94,\n",
       "      59,\n",
       "      30,\n",
       "      19,\n",
       "      11,\n",
       "      2,\n",
       "      4],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 13},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      8,\n",
       "      13,\n",
       "      25,\n",
       "      29,\n",
       "      15,\n",
       "      10,\n",
       "      9,\n",
       "      8,\n",
       "      3,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeQuality': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [42,\n",
       "      34,\n",
       "      79,\n",
       "      100,\n",
       "      20247,\n",
       "      14345,\n",
       "      9795,\n",
       "      6614,\n",
       "      5441,\n",
       "      3845,\n",
       "      2497,\n",
       "      2219,\n",
       "      1635,\n",
       "      1101,\n",
       "      926,\n",
       "      674,\n",
       "      461,\n",
       "      380,\n",
       "      262,\n",
       "      833],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      1,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      119],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'siteQualityMetrics': [{'metric': 'FS', 'value': 1.1223795673810724},\n",
       "    {'metric': 'InbreedingCoeff', 'value': -0.0008395130824121466},\n",
       "    {'metric': 'MQ', 'value': 59.99216274475224},\n",
       "    {'metric': 'MQRankSum', 'value': 0},\n",
       "    {'metric': 'QD', 'value': 13.505564},\n",
       "    {'metric': 'ReadPosRankSum', 'value': 0.345},\n",
       "    {'metric': 'SiteQuality', 'value': 100738},\n",
       "    {'metric': 'SOR', 'value': 0.804},\n",
       "    {'metric': 'VQSLOD', 'value': None}]}},\n",
       " 'flags': [],\n",
       " 'rsid': 'rs121965019',\n",
       " 'sortedTranscriptConsequences': [{'canonical': True,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'p.Trp402Ter',\n",
       "   'hgvsc': 'c.1205G>A',\n",
       "   'hgvsp': 'p.Trp402Ter',\n",
       "   'lof': 'HC',\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'stop_gained',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000247933',\n",
       "   'transcript_version': '8'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'p.Trp270Ter',\n",
       "   'hgvsc': 'c.809G>A',\n",
       "   'hgvsp': 'p.Trp270Ter',\n",
       "   'lof': 'HC',\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'stop_gained',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000514224',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.7G>A',\n",
       "   'hgvsc': 'n.7G>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000502829',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.1312G>A',\n",
       "   'hgvsc': 'n.1312G>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000514698',\n",
       "   'transcript_version': '5'}]}"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "js['data']['variant']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'id': 'AFR', 'ac': 13, 'an': 41628, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'AFR_FEMALE', 'ac': 6, 'an': 22480, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'AFR_MALE', 'ac': 7, 'an': 19148, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'AMI_FEMALE', 'ac': 0, 'an': 470, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'AMI_MALE', 'ac': 0, 'an': 428, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'AMR', 'ac': 3, 'an': 13600, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'AMR_FEMALE', 'ac': 2, 'an': 5882, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'AMR_MALE', 'ac': 1, 'an': 7718, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'ASJ', 'ac': 0, 'an': 3316, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'ASJ_FEMALE', 'ac': 0, 'an': 1756, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'ASJ_MALE', 'ac': 0, 'an': 1560, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'EAS', 'ac': 0, 'an': 3124, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'EAS_FEMALE', 'ac': 0, 'an': 1440, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'EAS_MALE', 'ac': 0, 'an': 1684, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'FIN', 'ac': 3, 'an': 10322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'FIN_FEMALE', 'ac': 1, 'an': 2476, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'FIN_MALE', 'ac': 2, 'an': 7846, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'NFE', 'ac': 100, 'an': 64398, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'NFE_FEMALE', 'ac': 59, 'an': 37292, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'NFE_MALE', 'ac': 41, 'an': 27106, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'OTH', 'ac': 1, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'OTH_FEMALE', 'ac': 1, 'an': 1094, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'OTH_MALE', 'ac': 0, 'an': 1048, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'SAS', 'ac': 0, 'an': 3036, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'SAS_FEMALE', 'ac': 0, 'an': 560, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'SAS_MALE', 'ac': 0, 'an': 2476, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'FEMALE', 'ac': 69, 'an': 73450, 'ac_hemi': 0, 'ac_hom': 0},\n",
       " {'id': 'MALE', 'ac': 51, 'an': 69014, 'ac_hemi': 0, 'ac_hom': 0}]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "js['data']['variant']['genome'].pop('populations')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'variantId': '4-1002747-G-A',\n",
       " 'reference_genome': 'GRCh38',\n",
       " 'chrom': '4',\n",
       " 'pos': 1002747,\n",
       " 'ref': 'G',\n",
       " 'alt': 'A',\n",
       " 'colocatedVariants': ['4-1002747-G-T'],\n",
       " 'multiNucleotideVariants': [],\n",
       " 'exome': None,\n",
       " 'genome': {'ac': 120,\n",
       "  'an': 142464,\n",
       "  'ac_hemi': 0,\n",
       "  'ac_hom': 0,\n",
       "  'faf95': {'popmax': 0.0013063199999999997, 'popmax_population': 'NFE'},\n",
       "  'filters': [],\n",
       "  'age_distribution': None,\n",
       "  'qualityMetrics': {'alleleBalance': {'alt': {'bin_edges': [0,\n",
       "      0.05,\n",
       "      0.1,\n",
       "      0.15,\n",
       "      0.2,\n",
       "      0.25,\n",
       "      0.3,\n",
       "      0.35,\n",
       "      0.4,\n",
       "      0.45,\n",
       "      0.5,\n",
       "      0.55,\n",
       "      0.6,\n",
       "      0.65,\n",
       "      0.7,\n",
       "      0.75,\n",
       "      0.8,\n",
       "      0.85,\n",
       "      0.9,\n",
       "      0.95,\n",
       "      1],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      3,\n",
       "      3,\n",
       "      8,\n",
       "      26,\n",
       "      16,\n",
       "      26,\n",
       "      17,\n",
       "      13,\n",
       "      5,\n",
       "      3,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeDepth': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [11,\n",
       "      67,\n",
       "      641,\n",
       "      3138,\n",
       "      6868,\n",
       "      12475,\n",
       "      21477,\n",
       "      17174,\n",
       "      5825,\n",
       "      1989,\n",
       "      945,\n",
       "      476,\n",
       "      212,\n",
       "      94,\n",
       "      59,\n",
       "      30,\n",
       "      19,\n",
       "      11,\n",
       "      2,\n",
       "      4],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 13},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      8,\n",
       "      13,\n",
       "      25,\n",
       "      29,\n",
       "      15,\n",
       "      10,\n",
       "      9,\n",
       "      8,\n",
       "      3,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeQuality': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [42,\n",
       "      34,\n",
       "      79,\n",
       "      100,\n",
       "      20247,\n",
       "      14345,\n",
       "      9795,\n",
       "      6614,\n",
       "      5441,\n",
       "      3845,\n",
       "      2497,\n",
       "      2219,\n",
       "      1635,\n",
       "      1101,\n",
       "      926,\n",
       "      674,\n",
       "      461,\n",
       "      380,\n",
       "      262,\n",
       "      833],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      1,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      119],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'siteQualityMetrics': [{'metric': 'FS', 'value': 1.1223795673810724},\n",
       "    {'metric': 'InbreedingCoeff', 'value': -0.0008395130824121466},\n",
       "    {'metric': 'MQ', 'value': 59.99216274475224},\n",
       "    {'metric': 'MQRankSum', 'value': 0},\n",
       "    {'metric': 'QD', 'value': 13.505564},\n",
       "    {'metric': 'ReadPosRankSum', 'value': 0.345},\n",
       "    {'metric': 'SiteQuality', 'value': 100738},\n",
       "    {'metric': 'SOR', 'value': 0.804},\n",
       "    {'metric': 'VQSLOD', 'value': None}]}},\n",
       " 'flags': [],\n",
       " 'rsid': 'rs121965019',\n",
       " 'sortedTranscriptConsequences': [{'canonical': True,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'p.Trp402Ter',\n",
       "   'hgvsc': 'c.1205G>A',\n",
       "   'hgvsp': 'p.Trp402Ter',\n",
       "   'lof': 'HC',\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'stop_gained',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000247933',\n",
       "   'transcript_version': '8'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'p.Trp270Ter',\n",
       "   'hgvsc': 'c.809G>A',\n",
       "   'hgvsp': 'p.Trp270Ter',\n",
       "   'lof': 'HC',\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'stop_gained',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000514224',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.7G>A',\n",
       "   'hgvsc': 'n.7G>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000502829',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.1312G>A',\n",
       "   'hgvsc': 'n.1312G>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000514698',\n",
       "   'transcript_version': '5'}]}"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "js['data']['variant']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'4'"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "js['data']['variant']['chrom']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
